---
title: "Klein Benchmark"
author: "Quan and Michael "
date: "7/9/2018"
output: html_document
---


#using klein dataset
```{r}
#Load all the required packages
devtools::load_all("C:/Users/s5125601/Desktop/scGPS-master")
library(scater)
library(scran)

#Save the path to the dataset
path_to_data = "/Users/s5125601/Documents/BenchMarking/Datasets/KleinDataSet/"

#Read the data (From local drive or by URL) (If needed change to http)
klein_dat <- readRDS(paste0(path_to_data,"klein.rds"))
#klein_dat <- readRDS(url("https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/klein.rds"))

#Remove features with no gene expression
keep_features <- rowSums(counts(klein_dat) > 0) > 0
table(keep_features)
klein_dat <- klein_dat[keep_features, ]
dim(klein_dat)

#Use scran normalisation
klein_dat <- computeSumFactors(klein_dat)
klein_dat <- normalise(klein_dat)

#Create a count per million assay
cpm(klein_dat) <- calculateCPM(klein_dat)

#Remove spikes
is.spike <-grepl("^ERCC", rownames(klein_dat))
table(is.spike)
klein_dat <- klein_dat[!is.spike, ]

#Start the time here
start_time <- Sys.time()

#Extract the needed variables
klein_dat_exprs <- assays(klein_dat)[["cpm"]]
klein_dat_cellnames <- colnames(klein_dat)
klein_dat_cellnames <- data.frame("cellBarcodes" = klein_dat_cellnames)
klein_dat_GeneMetaData <- rownames(klein_dat)
klein_dat_GeneMetaData <- data.frame("GeneSymbol" = klein_dat_GeneMetaData)

#Store Data in scGPS format
mixedpop <-NewscGPS_SME(ExpressionMatrix = klein_dat_exprs, GeneMetadata = klein_dat_GeneMetaData, CellMetadata = klein_dat_cellnames)

#Cluster and plot data using SCORE
CORE_cluster_bagging <- CORE_scGPS_bagging(mixedpop, remove_outlier = c(0), PCA=FALSE, bagging_run = 30, subsample_proportion = .8)
plot_CORE(CORE_cluster_bagging$tree, list_clusters = CORE_cluster_bagging$Cluster)
plot_optimal_CORE(original_tree= CORE_cluster_bagging$tree, optimal_cluster = unlist(CORE_cluster_bagging$Cluster[CORE_cluster_bagging$optimal_index]), shift = -100)

#Stop the time here
end_time <- Sys.time()
time_difference_SCORE <- end_time - start_time

#Find data needed for comparisons and store in data frame
cell_types1 <- colData(klein_dat)$cell_type1
label_list <- unlist(klein_dat_cellnames$cellBarcodes)
cluster_list <- unlist(CORE_cluster_bagging$Cluster[CORE_cluster_bagging$optimal_index])
compare_frame <- data.frame("Gene_label" = label_list, "type1" = cell_types1,"cluster" = cluster_list)

#Save the results to a spreadsheet (requires perl)
library("WriteXLS")
WriteXLS(compare_frame, ExcelFileName = "C:/Users/s5125601/Desktop/SCORE.xlsx")

#Find the adjusted rand index
AdjustedRandIndex_SCORE <- mclust::adjustedRandIndex(compare_frame$type1, compare_frame$cluster)
estimated_k_SCORE <- CORE_cluster_bagging$optimalMax
HighResRand <- mclust::adjustedRandIndex(compare_frame$type1, unlist(CORE_cluster_bagging$Cluster[1]))

#Remove clutter from the environment
rm(list = setdiff(ls(), c("AdjustedRandIndex_SCORE", "time_difference_SCORE", "estimated_k_SCORE", "HighResRand")))
```



#Using SC3 to test the klein dataset

```{r}
#Load everything for SC3
library(SC3)
library(scater)

#Get the path to the dataset
path_to_data = "/Users/s5125601/Documents/BenchMarking/Datasets/KleinDataSet/"

#Read the data
sce <- readRDS(paste0(path_to_data,"klein.rds"))
#klein_dat <- readRDS(url("https://scrnaseq-public-datasets.s3.amazonaws.com/scater-objects/klein.rds"))

#Find the genes with all zero entries and remove
keep_features <- rowSums(counts(sce) > 0) > 0
sce <- sce[keep_features, ]

#Remove the spikes
is.spike <-grepl("^ERCC", rownames(sce))
table(is.spike)
sce <- sce[!is.spike, ]

#Start the time here
start_time <- Sys.time()

#Run sc3 with an estimation for k
sce <- sc3_prepare(sce, n_cores = 1, gene_filter = TRUE)
sce <- sc3_estimate_k(sce)
SC3_k_estimate <- as.integer(unlist(metadata(sce)$sc3))
sce <- sc3(sce, ks = SC3_k_estimate, biology = FALSE, gene_filter = TRUE, n_cores = 1, kmeans_nstart = 50)#Here we changed to 50 as there are over 2000 cells

#Stop the time here
end_time <- Sys.time()
time_difference_SC3 <- end_time - start_time

#Make a dataframe with the results we want to examine
cell_types1 <- colData(sce)$cell_type1
label_list <- rownames(colData(sce))
cluster_list <- as.numeric(colData(sce)[, paste0("sc3_", SC3_k_estimate, "_clusters")])
compare_frame <- data.frame("Gene_label" = label_list, "type1" = cell_types1, "cluster" = cluster_list)

#Save the results to a spreadsheet (requires perl)
library("WriteXLS")
WriteXLS(compare_frame, ExcelFileName = "C:/Users/s5125601/Desktop/SC3.xlsx")

#Find the Adjusted Rand Index
AdjustedRandIndex_SC3 <- mclust::adjustedRandIndex(compare_frame$type1, compare_frame$cluster)

#Remove unwanted data
rm(list = setdiff(ls(), c("AdjustedRandIndex_SC3", "time_difference_SC3", "SC3_k_estimate")))

```





