---
title: "scGPS introduction"
author: "Quan Nguyen"
date: "`r Sys.Date()`"
output: html_document
vignette: >
  %\VignetteIndexEntry{scGPS introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```
################################################################################
##Installation instruction
################################################################################

```{r, eval = FALSE}

#Prior to install scGPS you need to install the SummarizedExperiment bioconductor package as the following
#source("https://bioconductor.org/biocLite.R")
#biocLite("SummarizedExperiment")

#R/3.4.1 or above is required 

#To install from github (Depending on the configuration of the local computer or HPC, possible custom C++ compilation may be required - see installation trouble-shootings below) 
devtools::install_github("IMB-Computational-Genomics-Lab/scGPS")

#for C++ compilation trouble-shooting, manual download and install can be done from github
git clone https://github.com/IMB-Computational-Genomics-Lab/scGPS

#then check in scGPS/src if any of the precompiled (e.g. those with *.so and *.o) files exist and delete them before recompiling

#create a Makevars file in the  scGPS/src with one line: PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)

#then with the scGPS as the R working directory manually compile in R, using devtools
devools::document() 

#and then you can load the package: 
devtools::load_all()

```
##A simple workflow of the scGPS: given a mixed population with known subpopulations, estimate transition scores between these subpopulation 

```{r, warning = FALSE, message = FALSE}
devtools::load_all()

#load mixed population 1 (loaded from sample1 dataset, named it as day2)
day2 <- sample1
mixedpop1 <-NewscGPS_SME(ExpressionMatrix = day2$dat2_counts, GeneMetadata = day2$dat2geneInfo,
                     CellMetadata = day2$dat2_clusters)
#load mixed population 2 (loaded from sample2 dataset, named it as day5)
day5 <- sample2
mixedpop2 <-NewscGPS_SME(ExpressionMatrix = day5$dat5_counts, GeneMetadata = day5$dat5geneInfo,
                     CellMetadata = day5$dat5_clusters)
#load gene list (this can be any lists of user selected genes)
genes <-GeneList
genes <-genes$Merged_unique

#select a subpopulation
c_selectID <- 1

#run the test bootstrap
suppressWarnings(LSOLDA_dat <- bootstrap_scGPS(nboots = 2,mixedpop1 = mixedpop1, mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list()))

#display some results 
names(LSOLDA_dat)
LSOLDA_dat$LassoPredict
LSOLDA_dat$LDAPredict

#summary results LDA
summary_prediction_lda(LSOLDA_dat=LSOLDA_dat, nPredSubpop = 4)

#summary results Lasso
summary_prediction_lasso(LSOLDA_dat=LSOLDA_dat, nPredSubpop = 4)

#summary deviance 
summary_deviance(LSOLDA_dat)

```
################################################################################
##A complete workflow of the scGPS: given an unknown mixed population, find clusters and estimate relationship between clusters 
################################################################################

```{r, warning = FALSE, message = FALSE}
#given a single cell expression matrix, without clustering information 
day5 <- sample2
cellnames <- colnames(day5$dat5_counts)
cluster <-day5$dat5_clusters
cellnames <-data.frame("Cluster"=cluster, "cellBarcodes" = cellnames)

mixedpop2 <-NewscGPS_SME(ExpressionMatrix = day5$dat5_counts, GeneMetadata = day5$dat5geneInfo, CellMetadata = cellnames ) 

#let's find the clusters
CORE_cluster <- CORE_scGPS(mixedpop2, remove_outlier = c(0))
#let's plot all clusters
plot_CORE(CORE_cluster$tree, CORE_cluster$Cluster)

#let's plot just the optimal clustering result (with colored dendrogram)
optimal_index = which(CORE_cluster$optimalClust$KeyStats$Height == CORE_cluster$optimalClust$OptimalRes)

plot_optimal_CORE(original_tree= CORE_cluster$tree, optimal_cluster = unlist(CORE_cluster$Cluster[optimal_index]), shift = -200)

#let's compare with other dimensional reduction methods 
library(cidr)
t <- CIDR_scGPS(expression.matrix=assay(mixedpop2))
p2 <-plotReduced_scGPS(t, color_fac = factor(colData(mixedpop2)[,1]),palletes =1:length(unique(colData(mixedpop2)[,1])))
#may need to turn off the R graphic device dev.off() before plotting the following
p2

#load gene list (this can be any lists of user selected genes)
genes <-GeneList
genes <-genes$Merged_unique

#the gene list can also be generated objectively by differential expression analysis
#if the cluster information is in the mixedpop2 object, run this (if not run the CORE
#as described below) 
DEgenes <- findMarkers_scGPS(expression_matrix=assay(mixedpop2), cluster = mixedpop2@colData$Cluster)
names(DEgenes)

#users need to check the format of the gene input to make sure they are consistent to 
#the gene names in the expression matrix 

#add the CORE cluster information into the scGPS object 
Optimal_index <- which( CORE_cluster$optimalClust$KeyStats$Height == CORE_cluster$optimalClust$OptimalRes)
mixedpop2@colData$Cluster <- unlist(CORE_cluster$Cluster[[Optimal_index]])

#select a subpopulation
c_selectID <- 1

#run the test bootstrap with nboots = 2 runs

suppressWarnings(LSOLDA_dat <- bootstrap_scGPS(nboots = 2,mixedpop1 = mixedpop2, mixedpop2 = mixedpop2, genes=genes, c_selectID, listData =list()))

#display summary results 
#summary results LDA
row_cluster <-length(unique(mixedpop2@colData$Cluster))
summary_prediction_lda(LSOLDA_dat=LSOLDA_dat, nPredSubpop = row_cluster )

#summary results Lasso
summary_prediction_lasso(LSOLDA_dat=LSOLDA_dat, nPredSubpop = row_cluster)

#summary deviance 
summary_deviance(LSOLDA_dat)

```
