<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scGPS introduction â€¢ scGPS</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="scGPS introduction">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scGPS</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette.html">scGPS introduction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="vignette_files/htmlwidgets-1.2/htmlwidgets.js"></script><script src="vignette_files/d3-4.5.0/d3.min.js"></script><script src="vignette_files/sankey-1/sankey.js"></script><script src="vignette_files/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>scGPS introduction</h1>
                        <h4 class="author">Quan Nguyen</h4>
            
            <h4 class="date">2018-05-05</h4>
      
      
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </div>

    
    
<div id="installation-instruction" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-instruction" class="anchor"></a>1. Installation instruction</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Prior to installing scGPS you need to install the SummarizedExperiment</span>
<span class="co"># bioconductor package as the following</span>
<span class="co"># source('https://bioconductor.org/biocLite.R') biocLite('SummarizedExperiment')</span>

<span class="co"># To install scGPS from github (Depending on the configuration of the local</span>
<span class="co"># computer or HPC, possible custom C++ compilation may be required - see</span>
<span class="co"># installation trouble-shootings below)</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"IMB-Computational-Genomics-Lab/scGPS"</span>)

<span class="co"># for C++ compilation trouble-shooting, manual download and installation can be</span>
<span class="co"># done from github</span>

git clone https<span class="op">:</span><span class="er">//</span>github.com<span class="op">/</span>IMB<span class="op">-</span>Computational<span class="op">-</span>Genomics<span class="op">-</span>Lab<span class="op">/</span>scGPS

<span class="co"># then check in scGPS/src if any of the precompiled (e.g.  those with *.so and</span>
<span class="co"># *.o) files exist and delete them before recompiling</span>

<span class="co"># create a Makevars file in the scGPS/src with one line: PKG_LIBS =</span>
<span class="co"># $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)</span>

<span class="co"># then with the scGPS as the R working directory, manually recompile scGPS in R</span>
<span class="co"># using devtools to load and install functions</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/document">document</a></span>()
<span class="co"># update the NAMESPACE using the update_NAMESPACE.sh </span>
sh update_NAMESPACE.sh 
<span class="co">#for window system, to update the NAMESPACE: copy and paste the content of the file NAMESPACE_toAdd_cpp_links to end of the file NAMESPACE </span>

<span class="co">#load the package to the workspace </span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>()</code></pre></div>
</div>
<div id="a-simple-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-simple-workflow-of-the-scgps" class="anchor"></a>2. A simple workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: given a mixed population with known subpopulations, estimate transition scores between these subpopulation</em></p>
<div id="setup-scgps-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#setup-scgps-objects" class="anchor"></a>2.1 Setup scGPS objects</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># load mixed population 1 (loaded from sample1 dataset, named it as day2)</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>()

day2 &lt;-<span class="st"> </span>sample1

mixedpop1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/NewscGPS_SME.html">NewscGPS_SME</a></span>(<span class="dt">ExpressionMatrix =</span> day2<span class="op">$</span>dat2_counts, <span class="dt">GeneMetadata =</span> day2<span class="op">$</span>dat2geneInfo, 
    <span class="dt">CellMetadata =</span> day2<span class="op">$</span>dat2_clusters)

<span class="co"># load mixed population 2 (loaded from sample2 dataset, named it as day5)</span>
day5 &lt;-<span class="st"> </span>sample2
mixedpop2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/NewscGPS_SME.html">NewscGPS_SME</a></span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, 
    <span class="dt">CellMetadata =</span> day5<span class="op">$</span>dat5_clusters)

<span class="co"># load gene list (this can be any lists of user selected genes)</span>
genes &lt;-<span class="st"> </span>GeneList
genes &lt;-<span class="st"> </span>genes<span class="op">$</span>Merged_unique

<span class="co"># select a subpopulation</span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span></code></pre></div>
</div>
<div id="run-predictions" class="section level2">
<h2 class="hasAnchor">
<a href="#run-predictions" class="anchor"></a>2.2 Run predictions</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># run the test bootstrap</span>

LSOLDA_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_scGPS.html">bootstrap_scGPS</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, c_selectID, <span class="dt">listData =</span> <span class="kw">list</span>())
<span class="co">#&gt; Warning: package 'caret' was built under R version 3.4.4</span>
<span class="co">#&gt; Loading required package: lattice</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 48 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold03.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  5 29 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold06.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 11 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold07.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 91 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold08.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 56 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold10.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 16 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold03.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 29 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold05.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 11 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold06.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 91 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold07.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 16 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold09.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  5 48 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 16 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold04.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 11 29 91 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold07.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  5 72 appear to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold10.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 48 appears to be constant within groups</span>
<span class="co">#&gt; Warning in nominalTrainWorkflow(x = x, y = y, wts = weights, info =</span>
<span class="co">#&gt; trainInfo, : There were missing values in resampled performance measures.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = t(predictor_S1), y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev    Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.563e-15 2.647e-01</span>
<span class="co">#&gt;   [2,]  1  3.445e-02 2.412e-01</span>
<span class="co">#&gt;   [3,]  2  7.286e-02 2.198e-01</span>
<span class="co">#&gt;   [4,]  3  1.150e-01 2.002e-01</span>
<span class="co">#&gt;   [5,]  3  1.547e-01 1.825e-01</span>
<span class="co">#&gt;   [6,]  3  1.892e-01 1.662e-01</span>
<span class="co">#&gt;   [7,]  6  2.233e-01 1.515e-01</span>
<span class="co">#&gt;   [8,]  7  2.576e-01 1.380e-01</span>
<span class="co">#&gt;   [9,]  7  2.907e-01 1.258e-01</span>
<span class="co">#&gt;  [10,]  7  3.203e-01 1.146e-01</span>
<span class="co">#&gt;  [11,]  7  3.470e-01 1.044e-01</span>
<span class="co">#&gt;  [12,]  8  3.728e-01 9.513e-02</span>
<span class="co">#&gt;  [13,]  9  3.978e-01 8.668e-02</span>
<span class="co">#&gt;  [14,] 10  4.224e-01 7.898e-02</span>
<span class="co">#&gt;  [15,] 10  4.450e-01 7.196e-02</span>
<span class="co">#&gt;  [16,] 10  4.653e-01 6.557e-02</span>
<span class="co">#&gt;  [17,] 10  4.838e-01 5.974e-02</span>
<span class="co">#&gt;  [18,] 12  5.007e-01 5.444e-02</span>
<span class="co">#&gt;  [19,] 13  5.168e-01 4.960e-02</span>
<span class="co">#&gt;  [20,] 14  5.326e-01 4.519e-02</span>
<span class="co">#&gt;  [21,] 14  5.472e-01 4.118e-02</span>
<span class="co">#&gt;  [22,] 17  5.613e-01 3.752e-02</span>
<span class="co">#&gt;  [23,] 18  5.764e-01 3.419e-02</span>
<span class="co">#&gt;  [24,] 18  5.904e-01 3.115e-02</span>
<span class="co">#&gt;  [25,] 19  6.033e-01 2.838e-02</span>
<span class="co">#&gt;  [26,] 22  6.154e-01 2.586e-02</span>
<span class="co">#&gt;  [27,] 23  6.278e-01 2.356e-02</span>
<span class="co">#&gt;  [28,] 25  6.407e-01 2.147e-02</span>
<span class="co">#&gt;  [29,] 29  6.539e-01 1.956e-02</span>
<span class="co">#&gt;  [30,] 30  6.676e-01 1.783e-02</span>
<span class="co">#&gt;  [31,] 33  6.810e-01 1.624e-02</span>
<span class="co">#&gt;  [32,] 33  6.939e-01 1.480e-02</span>
<span class="co">#&gt;  [33,] 36  7.058e-01 1.348e-02</span>
<span class="co">#&gt;  [34,] 41  7.179e-01 1.229e-02</span>
<span class="co">#&gt;  [35,] 43  7.306e-01 1.120e-02</span>
<span class="co">#&gt;  [36,] 45  7.429e-01 1.020e-02</span>
<span class="co">#&gt;  [37,] 46  7.552e-01 9.294e-03</span>
<span class="co">#&gt;  [38,] 47  7.668e-01 8.469e-03</span>
<span class="co">#&gt;  [39,] 49  7.780e-01 7.716e-03</span>
<span class="co">#&gt;  [40,] 49  7.886e-01 7.031e-03</span>
<span class="co">#&gt;  [41,] 50  7.989e-01 6.406e-03</span>
<span class="co">#&gt;  [42,] 51  8.086e-01 5.837e-03</span>
<span class="co">#&gt;  [43,] 54  8.179e-01 5.319e-03</span>
<span class="co">#&gt;  [44,] 57  8.279e-01 4.846e-03</span>
<span class="co">#&gt;  [45,] 59  8.384e-01 4.416e-03</span>
<span class="co">#&gt;  [46,] 59  8.486e-01 4.023e-03</span>
<span class="co">#&gt;  [47,] 65  8.589e-01 3.666e-03</span>
<span class="co">#&gt;  [48,] 64  8.690e-01 3.340e-03</span>
<span class="co">#&gt;  [49,] 65  8.785e-01 3.043e-03</span>
<span class="co">#&gt;  [50,] 65  8.875e-01 2.773e-03</span>
<span class="co">#&gt;  [51,] 66  8.961e-01 2.527e-03</span>
<span class="co">#&gt;  [52,] 65  9.040e-01 2.302e-03</span>
<span class="co">#&gt;  [53,] 66  9.110e-01 2.098e-03</span>
<span class="co">#&gt;  [54,] 66  9.176e-01 1.911e-03</span>
<span class="co">#&gt;  [55,] 67  9.239e-01 1.742e-03</span>
<span class="co">#&gt;  [56,] 67  9.300e-01 1.587e-03</span>
<span class="co">#&gt;  [57,] 67  9.356e-01 1.446e-03</span>
<span class="co">#&gt;  [58,] 67  9.407e-01 1.317e-03</span>
<span class="co">#&gt;  [59,] 67  9.458e-01 1.200e-03</span>
<span class="co">#&gt;  [60,] 66  9.505e-01 1.094e-03</span>
<span class="co">#&gt;  [61,] 66  9.548e-01 9.966e-04</span>
<span class="co">#&gt;  [62,] 67  9.589e-01 9.081e-04</span>
<span class="co">#&gt;  [63,] 67  9.626e-01 8.274e-04</span>
<span class="co">#&gt;  [64,] 68  9.660e-01 7.539e-04</span>
<span class="co">#&gt;  [65,] 68  9.690e-01 6.869e-04</span>
<span class="co">#&gt;  [66,] 68  9.718e-01 6.259e-04</span>
<span class="co">#&gt;  [67,] 69  9.744e-01 5.703e-04</span>
<span class="co">#&gt;  [68,] 68  9.768e-01 5.196e-04</span>
<span class="co">#&gt;  [69,] 67  9.788e-01 4.735e-04</span>
<span class="co">#&gt;  [70,] 67  9.808e-01 4.314e-04</span>
<span class="co">#&gt;  [71,] 67  9.824e-01 3.931e-04</span>
<span class="co">#&gt;  [72,] 67  9.840e-01 3.582e-04</span>
<span class="co">#&gt;  [73,] 67  9.854e-01 3.263e-04</span>
<span class="co">#&gt;  [74,] 66  9.867e-01 2.974e-04</span>
<span class="co">#&gt;  [75,] 65  9.879e-01 2.709e-04</span>
<span class="co">#&gt;  [76,] 65  9.890e-01 2.469e-04</span>
<span class="co">#&gt;  [77,] 65  9.900e-01 2.249e-04</span>
<span class="co">#&gt;  [78,] 65  9.909e-01 2.050e-04</span>
<span class="co">#&gt;  [79,] 65  9.917e-01 1.867e-04</span>
<span class="co">#&gt;  [80,] 65  9.924e-01 1.702e-04</span>
<span class="co">#&gt;  [81,] 65  9.931e-01 1.550e-04</span>
<span class="co">#&gt;  [82,] 65  9.937e-01 1.413e-04</span>
<span class="co">#&gt;  [83,] 65  9.943e-01 1.287e-04</span>
<span class="co">#&gt;  [84,] 65  9.948e-01 1.173e-04</span>
<span class="co">#&gt;  [85,] 65  9.952e-01 1.069e-04</span>
<span class="co">#&gt;  [86,] 65  9.956e-01 9.737e-05</span>
<span class="co">#&gt;  [87,] 65  9.960e-01 8.872e-05</span>
<span class="co">#&gt;  [88,] 65  9.964e-01 8.084e-05</span>
<span class="co">#&gt;  [89,] 65  9.967e-01 7.366e-05</span>
<span class="co">#&gt;  [90,] 65  9.970e-01 6.711e-05</span>
<span class="co">#&gt;  [91,] 65  9.973e-01 6.115e-05</span>
<span class="co">#&gt;  [92,] 65  9.975e-01 5.572e-05</span>
<span class="co">#&gt;  [93,] 65  9.977e-01 5.077e-05</span>
<span class="co">#&gt;  [94,] 65  9.979e-01 4.626e-05</span>
<span class="co">#&gt;  [95,] 65  9.981e-01 4.215e-05</span>
<span class="co">#&gt;  [96,] 65  9.983e-01 3.840e-05</span>
<span class="co">#&gt;  [97,] 65  9.984e-01 3.499e-05</span>
<span class="co">#&gt;  [98,] 65  9.986e-01 3.188e-05</span>
<span class="co">#&gt;  [99,] 65  9.987e-01 2.905e-05</span>
<span class="co">#&gt; [100,] 65  9.988e-01 2.647e-05</span>
<span class="co">#&gt; [1] "done bootstrap 1"</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  3  9 56 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold02.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 12 24 45 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold03.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 5 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold05.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 78 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold07.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 15 79 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold09.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 21 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold10.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 83 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 83 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold04.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 21 56 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold05.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  5 12 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold06.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  9 15 78 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold07.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  3 10 24 45 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold08.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 79 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 45 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold02.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 12 24 78 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold04.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 21 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold05.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 56 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold07.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 79 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold08.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  5  9 15 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold09.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 3 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold10.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 83 appears to be constant within groups</span>
<span class="co">#&gt; Warning in nominalTrainWorkflow(x = x, y = y, wts = weights, info =</span>
<span class="co">#&gt; trainInfo, : There were missing values in resampled performance measures.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = t(predictor_S1), y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev    Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.563e-15 2.479e-01</span>
<span class="co">#&gt;   [2,]  1  3.029e-02 2.259e-01</span>
<span class="co">#&gt;   [3,]  2  6.301e-02 2.058e-01</span>
<span class="co">#&gt;   [4,]  3  9.748e-02 1.875e-01</span>
<span class="co">#&gt;   [5,]  4  1.304e-01 1.709e-01</span>
<span class="co">#&gt;   [6,]  4  1.608e-01 1.557e-01</span>
<span class="co">#&gt;   [7,]  4  1.872e-01 1.419e-01</span>
<span class="co">#&gt;   [8,]  6  2.118e-01 1.293e-01</span>
<span class="co">#&gt;   [9,]  6  2.360e-01 1.178e-01</span>
<span class="co">#&gt;  [10,]  6  2.574e-01 1.073e-01</span>
<span class="co">#&gt;  [11,]  7  2.773e-01 9.779e-02</span>
<span class="co">#&gt;  [12,]  8  2.977e-01 8.910e-02</span>
<span class="co">#&gt;  [13,]  9  3.165e-01 8.119e-02</span>
<span class="co">#&gt;  [14,] 12  3.363e-01 7.397e-02</span>
<span class="co">#&gt;  [15,] 13  3.561e-01 6.740e-02</span>
<span class="co">#&gt;  [16,] 14  3.752e-01 6.141e-02</span>
<span class="co">#&gt;  [17,] 16  3.933e-01 5.596e-02</span>
<span class="co">#&gt;  [18,] 16  4.108e-01 5.099e-02</span>
<span class="co">#&gt;  [19,] 18  4.277e-01 4.646e-02</span>
<span class="co">#&gt;  [20,] 20  4.474e-01 4.233e-02</span>
<span class="co">#&gt;  [21,] 22  4.671e-01 3.857e-02</span>
<span class="co">#&gt;  [22,] 25  4.857e-01 3.514e-02</span>
<span class="co">#&gt;  [23,] 29  5.039e-01 3.202e-02</span>
<span class="co">#&gt;  [24,] 32  5.226e-01 2.918e-02</span>
<span class="co">#&gt;  [25,] 34  5.406e-01 2.658e-02</span>
<span class="co">#&gt;  [26,] 34  5.580e-01 2.422e-02</span>
<span class="co">#&gt;  [27,] 35  5.741e-01 2.207e-02</span>
<span class="co">#&gt;  [28,] 36  5.889e-01 2.011e-02</span>
<span class="co">#&gt;  [29,] 37  6.033e-01 1.832e-02</span>
<span class="co">#&gt;  [30,] 40  6.176e-01 1.670e-02</span>
<span class="co">#&gt;  [31,] 41  6.312e-01 1.521e-02</span>
<span class="co">#&gt;  [32,] 43  6.442e-01 1.386e-02</span>
<span class="co">#&gt;  [33,] 43  6.563e-01 1.263e-02</span>
<span class="co">#&gt;  [34,] 45  6.677e-01 1.151e-02</span>
<span class="co">#&gt;  [35,] 47  6.787e-01 1.049e-02</span>
<span class="co">#&gt;  [36,] 49  6.898e-01 9.554e-03</span>
<span class="co">#&gt;  [37,] 51  7.011e-01 8.705e-03</span>
<span class="co">#&gt;  [38,] 52  7.121e-01 7.932e-03</span>
<span class="co">#&gt;  [39,] 52  7.224e-01 7.227e-03</span>
<span class="co">#&gt;  [40,] 54  7.322e-01 6.585e-03</span>
<span class="co">#&gt;  [41,] 59  7.429e-01 6.000e-03</span>
<span class="co">#&gt;  [42,] 60  7.539e-01 5.467e-03</span>
<span class="co">#&gt;  [43,] 61  7.642e-01 4.981e-03</span>
<span class="co">#&gt;  [44,] 63  7.744e-01 4.539e-03</span>
<span class="co">#&gt;  [45,] 67  7.856e-01 4.136e-03</span>
<span class="co">#&gt;  [46,] 66  7.975e-01 3.768e-03</span>
<span class="co">#&gt;  [47,] 65  8.091e-01 3.434e-03</span>
<span class="co">#&gt;  [48,] 67  8.210e-01 3.129e-03</span>
<span class="co">#&gt;  [49,] 67  8.328e-01 2.851e-03</span>
<span class="co">#&gt;  [50,] 68  8.443e-01 2.597e-03</span>
<span class="co">#&gt;  [51,] 68  8.548e-01 2.367e-03</span>
<span class="co">#&gt;  [52,] 67  8.648e-01 2.156e-03</span>
<span class="co">#&gt;  [53,] 67  8.741e-01 1.965e-03</span>
<span class="co">#&gt;  [54,] 67  8.829e-01 1.790e-03</span>
<span class="co">#&gt;  [55,] 68  8.912e-01 1.631e-03</span>
<span class="co">#&gt;  [56,] 69  8.991e-01 1.486e-03</span>
<span class="co">#&gt;  [57,] 69  9.067e-01 1.354e-03</span>
<span class="co">#&gt;  [58,] 69  9.143e-01 1.234e-03</span>
<span class="co">#&gt;  [59,] 68  9.214e-01 1.124e-03</span>
<span class="co">#&gt;  [60,] 70  9.281e-01 1.024e-03</span>
<span class="co">#&gt;  [61,] 70  9.344e-01 9.334e-04</span>
<span class="co">#&gt;  [62,] 72  9.401e-01 8.505e-04</span>
<span class="co">#&gt;  [63,] 73  9.454e-01 7.750e-04</span>
<span class="co">#&gt;  [64,] 74  9.503e-01 7.061e-04</span>
<span class="co">#&gt;  [65,] 74  9.546e-01 6.434e-04</span>
<span class="co">#&gt;  [66,] 74  9.587e-01 5.862e-04</span>
<span class="co">#&gt;  [67,] 74  9.624e-01 5.341e-04</span>
<span class="co">#&gt;  [68,] 73  9.658e-01 4.867e-04</span>
<span class="co">#&gt;  [69,] 73  9.688e-01 4.435e-04</span>
<span class="co">#&gt;  [70,] 73  9.716e-01 4.041e-04</span>
<span class="co">#&gt;  [71,] 73  9.741e-01 3.682e-04</span>
<span class="co">#&gt;  [72,] 73  9.765e-01 3.355e-04</span>
<span class="co">#&gt;  [73,] 73  9.785e-01 3.057e-04</span>
<span class="co">#&gt;  [74,] 74  9.804e-01 2.785e-04</span>
<span class="co">#&gt;  [75,] 74  9.822e-01 2.538e-04</span>
<span class="co">#&gt;  [76,] 75  9.837e-01 2.312e-04</span>
<span class="co">#&gt;  [77,] 74  9.852e-01 2.107e-04</span>
<span class="co">#&gt;  [78,] 74  9.865e-01 1.920e-04</span>
<span class="co">#&gt;  [79,] 74  9.877e-01 1.749e-04</span>
<span class="co">#&gt;  [80,] 74  9.888e-01 1.594e-04</span>
<span class="co">#&gt;  [81,] 74  9.898e-01 1.452e-04</span>
<span class="co">#&gt;  [82,] 74  9.907e-01 1.323e-04</span>
<span class="co">#&gt;  [83,] 73  9.915e-01 1.206e-04</span>
<span class="co">#&gt;  [84,] 73  9.922e-01 1.098e-04</span>
<span class="co">#&gt;  [85,] 73  9.929e-01 1.001e-04</span>
<span class="co">#&gt;  [86,] 73  9.935e-01 9.120e-05</span>
<span class="co">#&gt;  [87,] 73  9.941e-01 8.310e-05</span>
<span class="co">#&gt;  [88,] 73  9.946e-01 7.571e-05</span>
<span class="co">#&gt;  [89,] 73  9.951e-01 6.899e-05</span>
<span class="co">#&gt;  [90,] 73  9.955e-01 6.286e-05</span>
<span class="co">#&gt;  [91,] 73  9.959e-01 5.727e-05</span>
<span class="co">#&gt;  [92,] 73  9.963e-01 5.219e-05</span>
<span class="co">#&gt;  [93,] 74  9.966e-01 4.755e-05</span>
<span class="co">#&gt;  [94,] 74  9.969e-01 4.333e-05</span>
<span class="co">#&gt;  [95,] 74  9.972e-01 3.948e-05</span>
<span class="co">#&gt;  [96,] 74  9.974e-01 3.597e-05</span>
<span class="co">#&gt;  [97,] 74  9.976e-01 3.277e-05</span>
<span class="co">#&gt;  [98,] 73  9.978e-01 2.986e-05</span>
<span class="co">#&gt;  [99,] 73  9.980e-01 2.721e-05</span>
<span class="co">#&gt; [100,] 73  9.982e-01 2.479e-05</span>
<span class="co">#&gt; [1] "done bootstrap 2"</span></code></pre></div>
</div>
<div id="summarise-results" class="section level2">
<h2 class="hasAnchor">
<a href="#summarise-results" class="anchor"></a>2.3 Summarise results</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_scGPS.html">bootstrap_scGPS</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>, <span class="dt">mixedpop1 =</span> mixedpop1, 
    <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes =</span> genes, c_selectID, <span class="dt">listData =</span> <span class="kw">list</span>())
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  4 17 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold02.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 12 49 82 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold03.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 11 16 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold04.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 24 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold05.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 30 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold08.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 63 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold10.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 46 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 17 49 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold02.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 12 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold03.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 63 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold05.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 30 82 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold07.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 16 46 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold10.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  4 11 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 11 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold02.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 30 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold03.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 12 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold04.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 63 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold06.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 16 82 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold07.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 4 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold09.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 17 49 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold10.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 46 appears to be constant within groups</span>
<span class="co">#&gt; Warning in nominalTrainWorkflow(x = x, y = y, wts = weights, info =</span>
<span class="co">#&gt; trainInfo, : There were missing values in resampled performance measures.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = t(predictor_S1), y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev    Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.563e-15 3.021e-01</span>
<span class="co">#&gt;   [2,]  1  4.487e-02 2.753e-01</span>
<span class="co">#&gt;   [3,]  1  8.291e-02 2.508e-01</span>
<span class="co">#&gt;   [4,]  1  1.157e-01 2.285e-01</span>
<span class="co">#&gt;   [5,]  1  1.442e-01 2.082e-01</span>
<span class="co">#&gt;   [6,]  2  1.737e-01 1.897e-01</span>
<span class="co">#&gt;   [7,]  3  2.020e-01 1.729e-01</span>
<span class="co">#&gt;   [8,]  3  2.312e-01 1.575e-01</span>
<span class="co">#&gt;   [9,]  3  2.570e-01 1.435e-01</span>
<span class="co">#&gt;  [10,]  4  2.811e-01 1.308e-01</span>
<span class="co">#&gt;  [11,]  4  3.044e-01 1.192e-01</span>
<span class="co">#&gt;  [12,]  4  3.253e-01 1.086e-01</span>
<span class="co">#&gt;  [13,]  4  3.440e-01 9.892e-02</span>
<span class="co">#&gt;  [14,]  9  3.631e-01 9.014e-02</span>
<span class="co">#&gt;  [15,]  9  3.836e-01 8.213e-02</span>
<span class="co">#&gt;  [16,]  9  4.019e-01 7.483e-02</span>
<span class="co">#&gt;  [17,]  9  4.183e-01 6.818e-02</span>
<span class="co">#&gt;  [18,] 10  4.337e-01 6.213e-02</span>
<span class="co">#&gt;  [19,] 12  4.494e-01 5.661e-02</span>
<span class="co">#&gt;  [20,] 14  4.680e-01 5.158e-02</span>
<span class="co">#&gt;  [21,] 18  4.891e-01 4.700e-02</span>
<span class="co">#&gt;  [22,] 19  5.106e-01 4.282e-02</span>
<span class="co">#&gt;  [23,] 23  5.320e-01 3.902e-02</span>
<span class="co">#&gt;  [24,] 24  5.522e-01 3.555e-02</span>
<span class="co">#&gt;  [25,] 26  5.704e-01 3.239e-02</span>
<span class="co">#&gt;  [26,] 28  5.881e-01 2.952e-02</span>
<span class="co">#&gt;  [27,] 30  6.041e-01 2.689e-02</span>
<span class="co">#&gt;  [28,] 30  6.190e-01 2.450e-02</span>
<span class="co">#&gt;  [29,] 31  6.328e-01 2.233e-02</span>
<span class="co">#&gt;  [30,] 36  6.467e-01 2.034e-02</span>
<span class="co">#&gt;  [31,] 37  6.609e-01 1.854e-02</span>
<span class="co">#&gt;  [32,] 38  6.742e-01 1.689e-02</span>
<span class="co">#&gt;  [33,] 40  6.868e-01 1.539e-02</span>
<span class="co">#&gt;  [34,] 44  6.992e-01 1.402e-02</span>
<span class="co">#&gt;  [35,] 46  7.111e-01 1.278e-02</span>
<span class="co">#&gt;  [36,] 46  7.221e-01 1.164e-02</span>
<span class="co">#&gt;  [37,] 50  7.326e-01 1.061e-02</span>
<span class="co">#&gt;  [38,] 54  7.428e-01 9.665e-03</span>
<span class="co">#&gt;  [39,] 56  7.534e-01 8.806e-03</span>
<span class="co">#&gt;  [40,] 57  7.641e-01 8.024e-03</span>
<span class="co">#&gt;  [41,] 58  7.742e-01 7.311e-03</span>
<span class="co">#&gt;  [42,] 59  7.839e-01 6.662e-03</span>
<span class="co">#&gt;  [43,] 61  7.932e-01 6.070e-03</span>
<span class="co">#&gt;  [44,] 63  8.021e-01 5.531e-03</span>
<span class="co">#&gt;  [45,] 64  8.106e-01 5.039e-03</span>
<span class="co">#&gt;  [46,] 66  8.185e-01 4.592e-03</span>
<span class="co">#&gt;  [47,] 66  8.260e-01 4.184e-03</span>
<span class="co">#&gt;  [48,] 66  8.331e-01 3.812e-03</span>
<span class="co">#&gt;  [49,] 67  8.399e-01 3.473e-03</span>
<span class="co">#&gt;  [50,] 68  8.465e-01 3.165e-03</span>
<span class="co">#&gt;  [51,] 71  8.531e-01 2.884e-03</span>
<span class="co">#&gt;  [52,] 71  8.602e-01 2.628e-03</span>
<span class="co">#&gt;  [53,] 72  8.672e-01 2.394e-03</span>
<span class="co">#&gt;  [54,] 71  8.742e-01 2.181e-03</span>
<span class="co">#&gt;  [55,] 71  8.814e-01 1.988e-03</span>
<span class="co">#&gt;  [56,] 70  8.883e-01 1.811e-03</span>
<span class="co">#&gt;  [57,] 70  8.952e-01 1.650e-03</span>
<span class="co">#&gt;  [58,] 70  9.022e-01 1.504e-03</span>
<span class="co">#&gt;  [59,] 67  9.090e-01 1.370e-03</span>
<span class="co">#&gt;  [60,] 67  9.157e-01 1.248e-03</span>
<span class="co">#&gt;  [61,] 66  9.223e-01 1.137e-03</span>
<span class="co">#&gt;  [62,] 66  9.287e-01 1.036e-03</span>
<span class="co">#&gt;  [63,] 65  9.345e-01 9.443e-04</span>
<span class="co">#&gt;  [64,] 68  9.409e-01 8.604e-04</span>
<span class="co">#&gt;  [65,] 66  9.462e-01 7.840e-04</span>
<span class="co">#&gt;  [66,] 67  9.513e-01 7.143e-04</span>
<span class="co">#&gt;  [67,] 67  9.559e-01 6.509e-04</span>
<span class="co">#&gt;  [68,] 67  9.600e-01 5.930e-04</span>
<span class="co">#&gt;  [69,] 68  9.638e-01 5.404e-04</span>
<span class="co">#&gt;  [70,] 67  9.672e-01 4.923e-04</span>
<span class="co">#&gt;  [71,] 66  9.702e-01 4.486e-04</span>
<span class="co">#&gt;  [72,] 66  9.729e-01 4.088e-04</span>
<span class="co">#&gt;  [73,] 65  9.753e-01 3.724e-04</span>
<span class="co">#&gt;  [74,] 64  9.775e-01 3.394e-04</span>
<span class="co">#&gt;  [75,] 64  9.795e-01 3.092e-04</span>
<span class="co">#&gt;  [76,] 64  9.814e-01 2.817e-04</span>
<span class="co">#&gt;  [77,] 64  9.831e-01 2.567e-04</span>
<span class="co">#&gt;  [78,] 64  9.846e-01 2.339e-04</span>
<span class="co">#&gt;  [79,] 64  9.859e-01 2.131e-04</span>
<span class="co">#&gt;  [80,] 65  9.872e-01 1.942e-04</span>
<span class="co">#&gt;  [81,] 66  9.883e-01 1.769e-04</span>
<span class="co">#&gt;  [82,] 66  9.894e-01 1.612e-04</span>
<span class="co">#&gt;  [83,] 66  9.903e-01 1.469e-04</span>
<span class="co">#&gt;  [84,] 66  9.912e-01 1.338e-04</span>
<span class="co">#&gt;  [85,] 67  9.920e-01 1.220e-04</span>
<span class="co">#&gt;  [86,] 67  9.927e-01 1.111e-04</span>
<span class="co">#&gt;  [87,] 66  9.933e-01 1.013e-04</span>
<span class="co">#&gt;  [88,] 66  9.939e-01 9.226e-05</span>
<span class="co">#&gt;  [89,] 66  9.944e-01 8.406e-05</span>
<span class="co">#&gt;  [90,] 66  9.949e-01 7.659e-05</span>
<span class="co">#&gt;  [91,] 67  9.954e-01 6.979e-05</span>
<span class="co">#&gt;  [92,] 67  9.958e-01 6.359e-05</span>
<span class="co">#&gt;  [93,] 67  9.962e-01 5.794e-05</span>
<span class="co">#&gt;  [94,] 67  9.965e-01 5.279e-05</span>
<span class="co">#&gt;  [95,] 67  9.968e-01 4.810e-05</span>
<span class="co">#&gt;  [96,] 66  9.971e-01 4.383e-05</span>
<span class="co">#&gt;  [97,] 67  9.973e-01 3.994e-05</span>
<span class="co">#&gt;  [98,] 67  9.976e-01 3.639e-05</span>
<span class="co">#&gt;  [99,] 67  9.978e-01 3.316e-05</span>
<span class="co">#&gt; [100,] 67  9.980e-01 3.021e-05</span>
<span class="co">#&gt; [1] "done bootstrap 1"</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 13 18 50 82 appear to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold03.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 86 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold05.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 95 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold06.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 59 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>

<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold09.Rep1: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 1 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 13 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold03.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 18 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold04.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 47 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>

<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold07.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 59 86 appear to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold09.Rep2: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables  1 50 82 95 appear to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold01.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 13 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold02.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 86 appears to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold03.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 18 95 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold04.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 1 appears to be constant within groups</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>

<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>

<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>

<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; Warning: model fit failed for Fold09.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variables 50 82 appear to be constant within groups</span>
<span class="co">#&gt; Warning: model fit failed for Fold10.Rep3: parameter=none Error in lda.default(x, grouping, ...) : </span>
<span class="co">#&gt;   variable 59 appears to be constant within groups</span>
<span class="co">#&gt; Warning in nominalTrainWorkflow(x = x, y = y, wts = weights, info =</span>
<span class="co">#&gt; trainInfo, : There were missing values in resampled performance measures.</span>
<span class="co">#&gt; Warning in lda.default(x, grouping, ...): variables are collinear</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = t(predictor_S1), y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev    Lambda</span>
<span class="co">#&gt;   [1,]  0 -2.563e-15 2.603e-01</span>
<span class="co">#&gt;   [2,]  2  3.890e-02 2.371e-01</span>
<span class="co">#&gt;   [3,]  2  7.817e-02 2.161e-01</span>
<span class="co">#&gt;   [4,]  2  1.120e-01 1.969e-01</span>
<span class="co">#&gt;   [5,]  2  1.416e-01 1.794e-01</span>
<span class="co">#&gt;   [6,]  3  1.683e-01 1.635e-01</span>
<span class="co">#&gt;   [7,]  4  1.983e-01 1.489e-01</span>
<span class="co">#&gt;   [8,]  4  2.262e-01 1.357e-01</span>
<span class="co">#&gt;   [9,]  4  2.510e-01 1.237e-01</span>
<span class="co">#&gt;  [10,]  6  2.753e-01 1.127e-01</span>
<span class="co">#&gt;  [11,]  7  2.991e-01 1.027e-01</span>
<span class="co">#&gt;  [12,]  8  3.213e-01 9.354e-02</span>
<span class="co">#&gt;  [13,]  8  3.425e-01 8.523e-02</span>
<span class="co">#&gt;  [14,]  8  3.615e-01 7.766e-02</span>
<span class="co">#&gt;  [15,]  9  3.787e-01 7.076e-02</span>
<span class="co">#&gt;  [16,]  9  3.950e-01 6.447e-02</span>
<span class="co">#&gt;  [17,] 10  4.103e-01 5.874e-02</span>
<span class="co">#&gt;  [18,] 11  4.247e-01 5.353e-02</span>
<span class="co">#&gt;  [19,] 12  4.384e-01 4.877e-02</span>
<span class="co">#&gt;  [20,] 13  4.514e-01 4.444e-02</span>
<span class="co">#&gt;  [21,] 16  4.646e-01 4.049e-02</span>
<span class="co">#&gt;  [22,] 18  4.792e-01 3.689e-02</span>
<span class="co">#&gt;  [23,] 21  4.952e-01 3.362e-02</span>
<span class="co">#&gt;  [24,] 25  5.133e-01 3.063e-02</span>
<span class="co">#&gt;  [25,] 28  5.317e-01 2.791e-02</span>
<span class="co">#&gt;  [26,] 31  5.495e-01 2.543e-02</span>
<span class="co">#&gt;  [27,] 31  5.660e-01 2.317e-02</span>
<span class="co">#&gt;  [28,] 36  5.818e-01 2.111e-02</span>
<span class="co">#&gt;  [29,] 40  5.972e-01 1.924e-02</span>
<span class="co">#&gt;  [30,] 45  6.142e-01 1.753e-02</span>
<span class="co">#&gt;  [31,] 45  6.309e-01 1.597e-02</span>
<span class="co">#&gt;  [32,] 47  6.466e-01 1.455e-02</span>
<span class="co">#&gt;  [33,] 51  6.615e-01 1.326e-02</span>
<span class="co">#&gt;  [34,] 51  6.756e-01 1.208e-02</span>
<span class="co">#&gt;  [35,] 54  6.892e-01 1.101e-02</span>
<span class="co">#&gt;  [36,] 55  7.024e-01 1.003e-02</span>
<span class="co">#&gt;  [37,] 57  7.146e-01 9.139e-03</span>
<span class="co">#&gt;  [38,] 57  7.260e-01 8.327e-03</span>
<span class="co">#&gt;  [39,] 56  7.362e-01 7.587e-03</span>
<span class="co">#&gt;  [40,] 62  7.459e-01 6.913e-03</span>
<span class="co">#&gt;  [41,] 62  7.555e-01 6.299e-03</span>
<span class="co">#&gt;  [42,] 63  7.644e-01 5.739e-03</span>
<span class="co">#&gt;  [43,] 63  7.729e-01 5.229e-03</span>
<span class="co">#&gt;  [44,] 66  7.818e-01 4.765e-03</span>
<span class="co">#&gt;  [45,] 66  7.905e-01 4.342e-03</span>
<span class="co">#&gt;  [46,] 67  7.987e-01 3.956e-03</span>
<span class="co">#&gt;  [47,] 69  8.066e-01 3.604e-03</span>
<span class="co">#&gt;  [48,] 69  8.144e-01 3.284e-03</span>
<span class="co">#&gt;  [49,] 69  8.222e-01 2.992e-03</span>
<span class="co">#&gt;  [50,] 69  8.297e-01 2.727e-03</span>
<span class="co">#&gt;  [51,] 70  8.371e-01 2.484e-03</span>
<span class="co">#&gt;  [52,] 70  8.450e-01 2.264e-03</span>
<span class="co">#&gt;  [53,] 67  8.528e-01 2.063e-03</span>
<span class="co">#&gt;  [54,] 68  8.599e-01 1.879e-03</span>
<span class="co">#&gt;  [55,] 68  8.667e-01 1.712e-03</span>
<span class="co">#&gt;  [56,] 70  8.736e-01 1.560e-03</span>
<span class="co">#&gt;  [57,] 69  8.805e-01 1.422e-03</span>
<span class="co">#&gt;  [58,] 69  8.872e-01 1.295e-03</span>
<span class="co">#&gt;  [59,] 73  8.940e-01 1.180e-03</span>
<span class="co">#&gt;  [60,] 74  9.016e-01 1.075e-03</span>
<span class="co">#&gt;  [61,] 76  9.092e-01 9.799e-04</span>
<span class="co">#&gt;  [62,] 77  9.168e-01 8.929e-04</span>
<span class="co">#&gt;  [63,] 76  9.239e-01 8.135e-04</span>
<span class="co">#&gt;  [64,] 77  9.304e-01 7.413e-04</span>
<span class="co">#&gt;  [65,] 76  9.364e-01 6.754e-04</span>
<span class="co">#&gt;  [66,] 75  9.419e-01 6.154e-04</span>
<span class="co">#&gt;  [67,] 75  9.470e-01 5.607e-04</span>
<span class="co">#&gt;  [68,] 76  9.518e-01 5.109e-04</span>
<span class="co">#&gt;  [69,] 75  9.563e-01 4.655e-04</span>
<span class="co">#&gt;  [70,] 74  9.603e-01 4.242e-04</span>
<span class="co">#&gt;  [71,] 74  9.639e-01 3.865e-04</span>
<span class="co">#&gt;  [72,] 74  9.673e-01 3.522e-04</span>
<span class="co">#&gt;  [73,] 74  9.703e-01 3.209e-04</span>
<span class="co">#&gt;  [74,] 74  9.730e-01 2.924e-04</span>
<span class="co">#&gt;  [75,] 74  9.755e-01 2.664e-04</span>
<span class="co">#&gt;  [76,] 74  9.777e-01 2.427e-04</span>
<span class="co">#&gt;  [77,] 74  9.797e-01 2.212e-04</span>
<span class="co">#&gt;  [78,] 72  9.816e-01 2.015e-04</span>
<span class="co">#&gt;  [79,] 72  9.832e-01 1.836e-04</span>
<span class="co">#&gt;  [80,] 72  9.847e-01 1.673e-04</span>
<span class="co">#&gt;  [81,] 72  9.861e-01 1.524e-04</span>
<span class="co">#&gt;  [82,] 72  9.873e-01 1.389e-04</span>
<span class="co">#&gt;  [83,] 72  9.885e-01 1.266e-04</span>
<span class="co">#&gt;  [84,] 72  9.895e-01 1.153e-04</span>
<span class="co">#&gt;  [85,] 72  9.904e-01 1.051e-04</span>
<span class="co">#&gt;  [86,] 72  9.913e-01 9.574e-05</span>
<span class="co">#&gt;  [87,] 72  9.921e-01 8.723e-05</span>
<span class="co">#&gt;  [88,] 72  9.927e-01 7.948e-05</span>
<span class="co">#&gt;  [89,] 72  9.934e-01 7.242e-05</span>
<span class="co">#&gt;  [90,] 73  9.940e-01 6.599e-05</span>
<span class="co">#&gt;  [91,] 73  9.945e-01 6.013e-05</span>
<span class="co">#&gt;  [92,] 72  9.950e-01 5.478e-05</span>
<span class="co">#&gt;  [93,] 72  9.954e-01 4.992e-05</span>
<span class="co">#&gt;  [94,] 72  9.958e-01 4.548e-05</span>
<span class="co">#&gt;  [95,] 72  9.962e-01 4.144e-05</span>
<span class="co">#&gt;  [96,] 73  9.965e-01 3.776e-05</span>
<span class="co">#&gt;  [97,] 74  9.968e-01 3.441e-05</span>
<span class="co">#&gt;  [98,] 74  9.971e-01 3.135e-05</span>
<span class="co">#&gt;  [99,] 74  9.974e-01 2.856e-05</span>
<span class="co">#&gt; [100,] 74  9.976e-01 2.603e-05</span>
<span class="co">#&gt; [1] "done bootstrap 2"</span>

<span class="co"># display the list of result information in the LASOLDA_dat object </span>
<span class="kw">names</span>(LSOLDA_dat)
<span class="co">#&gt; [1] "Accuracy"     "LassoGenes"   "Deviance"     "LassoFit"    </span>
<span class="co">#&gt; [5] "LDAFit"       "predictor_S1" "LassoPredict" "LDAPredict"</span>
LSOLDA_dat<span class="op">$</span>LassoPredict
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; [[1]][[1]]</span>
<span class="co">#&gt; [1] "LASSO for subpop1 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[2]]</span>
<span class="co">#&gt; [1] 5.882353</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[3]]</span>
<span class="co">#&gt; [1] "LASSO for subpop2 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[4]]</span>
<span class="co">#&gt; [1] 12.85714</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[5]]</span>
<span class="co">#&gt; [1] "LASSO for subpop3 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[6]]</span>
<span class="co">#&gt; [1] 0.7518797</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[7]]</span>
<span class="co">#&gt; [1] "LASSO for subpop4 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[8]]</span>
<span class="co">#&gt; [1] 7.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt; [[2]][[1]]</span>
<span class="co">#&gt; [1] "LASSO for subpop1 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[2]]</span>
<span class="co">#&gt; [1] 89.83957</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[3]]</span>
<span class="co">#&gt; [1] "LASSO for subpop2 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[4]]</span>
<span class="co">#&gt; [1] 98.57143</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[5]]</span>
<span class="co">#&gt; [1] "LASSO for subpop3 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[6]]</span>
<span class="co">#&gt; [1] 70.67669</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[7]]</span>
<span class="co">#&gt; [1] "LASSO for subpop4 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[8]]</span>
<span class="co">#&gt; [1] 87.5</span>
LSOLDA_dat<span class="op">$</span>LDAPredict
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; [[1]][[1]]</span>
<span class="co">#&gt; [1] "LDA for subpop 1 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[2]]</span>
<span class="co">#&gt; [1] 3.208556</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[3]]</span>
<span class="co">#&gt; [1] "LDA for subpop 2 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[4]]</span>
<span class="co">#&gt; [1] 40.71429</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[5]]</span>
<span class="co">#&gt; [1] "LDA for subpop 3 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[6]]</span>
<span class="co">#&gt; [1] 1.503759</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[7]]</span>
<span class="co">#&gt; [1] "LDA for subpop 4 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[1]][[8]]</span>
<span class="co">#&gt; [1] 12.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt; [[2]][[1]]</span>
<span class="co">#&gt; [1] "LDA for subpop 1 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[2]]</span>
<span class="co">#&gt; [1] 52.94118</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[3]]</span>
<span class="co">#&gt; [1] "LDA for subpop 2 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[4]]</span>
<span class="co">#&gt; [1] 42.14286</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[5]]</span>
<span class="co">#&gt; [1] "LDA for subpop 3 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[6]]</span>
<span class="co">#&gt; [1] 47.36842</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[7]]</span>
<span class="co">#&gt; [1] "LDA for subpop 4 in target mixedpop2"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]][[8]]</span>
<span class="co">#&gt; [1] 40</span>

<span class="co"># summary results LDA</span>
<span class="kw"><a href="../reference/summary_prediction_lda.html">summary_prediction_lda</a></span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co">#&gt;                 V1               V2                                names</span>
<span class="co">#&gt; 1 3.20855614973262 52.9411764705882 LDA for subpop 1 in target mixedpop2</span>
<span class="co">#&gt; 2 40.7142857142857 42.1428571428571 LDA for subpop 2 in target mixedpop2</span>
<span class="co">#&gt; 3 1.50375939849624 47.3684210526316 LDA for subpop 3 in target mixedpop2</span>
<span class="co">#&gt; 4             12.5               40 LDA for subpop 4 in target mixedpop2</span>

<span class="co"># summary results Lasso</span>
<span class="kw"><a href="../reference/summary_prediction_lasso.html">summary_prediction_lasso</a></span>(<span class="dt">LSOLDA_dat =</span> LSOLDA_dat, <span class="dt">nPredSubpop =</span> <span class="dv">4</span>)
<span class="co">#&gt;                 V1               V2                                 names</span>
<span class="co">#&gt; 1 5.88235294117647 89.8395721925134 LASSO for subpop1 in target mixedpop2</span>
<span class="co">#&gt; 2 12.8571428571429 98.5714285714286 LASSO for subpop2 in target mixedpop2</span>
<span class="co">#&gt; 3 0.75187969924812 70.6766917293233 LASSO for subpop3 in target mixedpop2</span>
<span class="co">#&gt; 4              7.5             87.5 LASSO for subpop4 in target mixedpop2</span>

<span class="co"># summary deviance</span>
<span class="kw"><a href="../reference/summary_deviance.html">summary_deviance</a></span>(<span class="dt">object =</span> LSOLDA_dat)
<span class="co">#&gt; $allDeviance</span>
<span class="co">#&gt; [1] "0.344"  "0.4384"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $DeviMax</span>
<span class="co">#&gt;          Dfd   Deviance        DEgenes</span>
<span class="co">#&gt; 1          0 -2.563e-15 genes_cluster1</span>
<span class="co">#&gt; 2          2     0.1416 genes_cluster1</span>
<span class="co">#&gt; 3          3     0.1683 genes_cluster1</span>
<span class="co">#&gt; 4          4      0.251 genes_cluster1</span>
<span class="co">#&gt; 5          6     0.2753 genes_cluster1</span>
<span class="co">#&gt; 6          7     0.2991 genes_cluster1</span>
<span class="co">#&gt; 7          8     0.3615 genes_cluster1</span>
<span class="co">#&gt; 8          9      0.395 genes_cluster1</span>
<span class="co">#&gt; 9         10     0.4103 genes_cluster1</span>
<span class="co">#&gt; 10        11     0.4247 genes_cluster1</span>
<span class="co">#&gt; 11        12     0.4384 genes_cluster1</span>
<span class="co">#&gt; 12 remaining          1        DEgenes</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $LassoGenesMax</span>
<span class="co">#&gt;                                 1                  name</span>
<span class="co">#&gt; (Intercept)            0.15150185           (Intercept)</span>
<span class="co">#&gt; CXCR4_ENSG00000121966 -0.04667005 CXCR4_ENSG00000121966</span>
<span class="co">#&gt; MYO3B_ENSG00000071909 -0.03817082 MYO3B_ENSG00000071909</span>
<span class="co">#&gt; FN1_ENSG00000115414   -0.05540733   FN1_ENSG00000115414</span>
<span class="co">#&gt; ISL1_ENSG00000016082  -0.08555153  ISL1_ENSG00000016082</span>
<span class="co">#&gt; T_ENSG00000164458      0.13724407     T_ENSG00000164458</span>
<span class="co">#&gt; SOX17_ENSG00000164736 -0.02838181 SOX17_ENSG00000164736</span>
<span class="co">#&gt; MESP1_ENSG00000166823  0.07663741 MESP1_ENSG00000166823</span>
<span class="co">#&gt; MESP2_ENSG00000188095  0.01250023 MESP2_ENSG00000188095</span>
<span class="co">#&gt; FOXF1_ENSG00000103241  0.41199180 FOXF1_ENSG00000103241</span>
<span class="co">#&gt; HNF1B_ENSG00000275410 -0.16845583 HNF1B_ENSG00000275410</span>
<span class="co">#&gt; FOXA2_ENSG00000125798 -0.18449768 FOXA2_ENSG00000125798</span>
<span class="co">#&gt; SNAI1_ENSG00000124216  0.24078696 SNAI1_ENSG00000124216</span></code></pre></div>
</div>
</div>
<div id="a-complete-workflow-of-the-scgps" class="section level1">
<h1 class="hasAnchor">
<a href="#a-complete-workflow-of-the-scgps" class="anchor"></a>3. A complete workflow of the scGPS:</h1>
<p><em>The purpose of this workflow is to solve the following task: given an unknown mixed population, find clusters and estimate relationship between clusters</em></p>
<div id="identify-clusters-in-a-using-core" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters-in-a-using-core" class="anchor"></a>3.1 Identify clusters in a using CORE</h2>
<p><em>(skip this step if clusters are known)</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#Let's find clustering information in an expresion data</span>
day5 &lt;-<span class="st"> </span>sample2
cellnames &lt;-<span class="st"> </span><span class="kw">colnames</span>(day5<span class="op">$</span>dat5_counts)
cluster &lt;-day5<span class="op">$</span>dat5_clusters
cellnames &lt;-<span class="kw">data.frame</span>(<span class="st">"Cluster"</span>=cluster, <span class="st">"cellBarcodes"</span> =<span class="st"> </span>cellnames)
mixedpop2 &lt;-<span class="kw"><a href="../reference/NewscGPS_SME.html">NewscGPS_SME</a></span>(<span class="dt">ExpressionMatrix =</span> day5<span class="op">$</span>dat5_counts, <span class="dt">GeneMetadata =</span> day5<span class="op">$</span>dat5geneInfo, <span class="dt">CellMetadata =</span> cellnames ) 

CORE_cluster &lt;-<span class="st"> </span><span class="kw"><a href="../reference/CORE_scGPS.html">CORE_scGPS</a></span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw">c</span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; [1] "Identifying top variable genes"</span>
<span class="co">#&gt; [1] "Calculating distance matrix"</span>
<span class="co">#&gt; [1] "Performing hierarchical clustering"</span>
<span class="co">#&gt; [1] "Finding clustering information"</span>
<span class="co">#&gt; [1] "No more outliers detected after 1 filtering round"</span>
<span class="co">#&gt; [1] "writing clustering result for run 1"</span>
<span class="co">#&gt; [1] "writing clustering result for run 2"</span>
<span class="co">#&gt; [1] "writing clustering result for run 3"</span>
<span class="co">#&gt; [1] "writing clustering result for run 4"</span>
<span class="co">#&gt; [1] "writing clustering result for run 5"</span>
<span class="co">#&gt; [1] "writing clustering result for run 6"</span>
<span class="co">#&gt; [1] "writing clustering result for run 7"</span>
<span class="co">#&gt; [1] "writing clustering result for run 8"</span>
<span class="co">#&gt; [1] "writing clustering result for run 9"</span>
<span class="co">#&gt; [1] "writing clustering result for run 10"</span>
<span class="co">#&gt; [1] "writing clustering result for run 11"</span>
<span class="co">#&gt; [1] "writing clustering result for run 12"</span>
<span class="co">#&gt; [1] "writing clustering result for run 13"</span>
<span class="co">#&gt; [1] "writing clustering result for run 14"</span>
<span class="co">#&gt; [1] "writing clustering result for run 15"</span>
<span class="co">#&gt; [1] "writing clustering result for run 16"</span>
<span class="co">#&gt; [1] "writing clustering result for run 17"</span>
<span class="co">#&gt; [1] "writing clustering result for run 18"</span>
<span class="co">#&gt; [1] "writing clustering result for run 19"</span>
<span class="co">#&gt; [1] "writing clustering result for run 20"</span>
<span class="co">#&gt; [1] "writing clustering result for run 21"</span>
<span class="co">#&gt; [1] "writing clustering result for run 22"</span>
<span class="co">#&gt; [1] "writing clustering result for run 23"</span>
<span class="co">#&gt; [1] "writing clustering result for run 24"</span>
<span class="co">#&gt; [1] "writing clustering result for run 25"</span>
<span class="co">#&gt; [1] "writing clustering result for run 26"</span>
<span class="co">#&gt; [1] "writing clustering result for run 27"</span>
<span class="co">#&gt; [1] "writing clustering result for run 28"</span>
<span class="co">#&gt; [1] "writing clustering result for run 29"</span>
<span class="co">#&gt; [1] "writing clustering result for run 30"</span>
<span class="co">#&gt; [1] "writing clustering result for run 31"</span>
<span class="co">#&gt; [1] "writing clustering result for run 32"</span>
<span class="co">#&gt; [1] "writing clustering result for run 33"</span>
<span class="co">#&gt; [1] "writing clustering result for run 34"</span>
<span class="co">#&gt; [1] "writing clustering result for run 35"</span>
<span class="co">#&gt; [1] "writing clustering result for run 36"</span>
<span class="co">#&gt; [1] "writing clustering result for run 37"</span>
<span class="co">#&gt; [1] "writing clustering result for run 38"</span>
<span class="co">#&gt; [1] "writing clustering result for run 39"</span>
<span class="co">#&gt; [1] "writing clustering result for run 40"</span>
<span class="co">#&gt; [1] "Done clustering, moving to stability calculation..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done finding optimal clustering..."</span></code></pre></div>
</div>
<div id="visualise-all-cluster-results-in-all-iterations" class="section level2">
<h2 class="hasAnchor">
<a href="#visualise-all-cluster-results-in-all-iterations" class="anchor"></a>3.2 Visualise all cluster results in all iterations</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#plot with default colors</span>
<span class="kw"><a href="../reference/plot_CORE.html">plot_CORE</a></span>(CORE_cluster<span class="op">$</span>tree, CORE_cluster<span class="op">$</span>Cluster)

<span class="co">#let's find the CORE clusters</span>
CORE_cluster &lt;-<span class="st"> </span><span class="kw"><a href="../reference/CORE_scGPS.html">CORE_scGPS</a></span>(mixedpop2, <span class="dt">remove_outlier =</span> <span class="kw">c</span>(<span class="dv">0</span>), <span class="dt">PCA=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; [1] "Identifying top variable genes"</span>
<span class="co">#&gt; [1] "Calculating distance matrix"</span>
<span class="co">#&gt; [1] "Performing hierarchical clustering"</span>
<span class="co">#&gt; [1] "Finding clustering information"</span>
<span class="co">#&gt; [1] "No more outliers detected after 1 filtering round"</span>
<span class="co">#&gt; [1] "writing clustering result for run 1"</span>
<span class="co">#&gt; [1] "writing clustering result for run 2"</span>
<span class="co">#&gt; [1] "writing clustering result for run 3"</span>
<span class="co">#&gt; [1] "writing clustering result for run 4"</span>
<span class="co">#&gt; [1] "writing clustering result for run 5"</span>
<span class="co">#&gt; [1] "writing clustering result for run 6"</span>
<span class="co">#&gt; [1] "writing clustering result for run 7"</span>
<span class="co">#&gt; [1] "writing clustering result for run 8"</span>
<span class="co">#&gt; [1] "writing clustering result for run 9"</span>
<span class="co">#&gt; [1] "writing clustering result for run 10"</span>
<span class="co">#&gt; [1] "writing clustering result for run 11"</span>
<span class="co">#&gt; [1] "writing clustering result for run 12"</span>
<span class="co">#&gt; [1] "writing clustering result for run 13"</span>
<span class="co">#&gt; [1] "writing clustering result for run 14"</span>
<span class="co">#&gt; [1] "writing clustering result for run 15"</span>
<span class="co">#&gt; [1] "writing clustering result for run 16"</span>
<span class="co">#&gt; [1] "writing clustering result for run 17"</span>
<span class="co">#&gt; [1] "writing clustering result for run 18"</span>
<span class="co">#&gt; [1] "writing clustering result for run 19"</span>
<span class="co">#&gt; [1] "writing clustering result for run 20"</span>
<span class="co">#&gt; [1] "writing clustering result for run 21"</span>
<span class="co">#&gt; [1] "writing clustering result for run 22"</span>
<span class="co">#&gt; [1] "writing clustering result for run 23"</span>
<span class="co">#&gt; [1] "writing clustering result for run 24"</span>
<span class="co">#&gt; [1] "writing clustering result for run 25"</span>
<span class="co">#&gt; [1] "writing clustering result for run 26"</span>
<span class="co">#&gt; [1] "writing clustering result for run 27"</span>
<span class="co">#&gt; [1] "writing clustering result for run 28"</span>
<span class="co">#&gt; [1] "writing clustering result for run 29"</span>
<span class="co">#&gt; [1] "writing clustering result for run 30"</span>
<span class="co">#&gt; [1] "writing clustering result for run 31"</span>
<span class="co">#&gt; [1] "writing clustering result for run 32"</span>
<span class="co">#&gt; [1] "writing clustering result for run 33"</span>
<span class="co">#&gt; [1] "writing clustering result for run 34"</span>
<span class="co">#&gt; [1] "writing clustering result for run 35"</span>
<span class="co">#&gt; [1] "writing clustering result for run 36"</span>
<span class="co">#&gt; [1] "writing clustering result for run 37"</span>
<span class="co">#&gt; [1] "writing clustering result for run 38"</span>
<span class="co">#&gt; [1] "writing clustering result for run 39"</span>
<span class="co">#&gt; [1] "writing clustering result for run 40"</span>
<span class="co">#&gt; [1] "Done clustering, moving to stability calculation..."</span>
<span class="co">#&gt; [1] "Done calculating stability..."</span>
<span class="co">#&gt; [1] "Start finding optimal clustering..."</span>
<span class="co">#&gt; [1] "Done finding optimal clustering..."</span>

<span class="co">#let's plot all clusters</span>
<span class="kw"><a href="../reference/plot_CORE.html">plot_CORE</a></span>(CORE_cluster<span class="op">$</span>tree, CORE_cluster<span class="op">$</span>Cluster)</code></pre></div>
<p><img src="man/figures/unnamed-chunk-6-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#you can customise the cluster color bars (provide color_branch values)</span>
<span class="kw"><a href="../reference/plot_CORE.html">plot_CORE</a></span>(CORE_cluster<span class="op">$</span>tree, CORE_cluster<span class="op">$</span>Cluster, <span class="dt">color_branch =</span> <span class="kw">c</span>(<span class="st">"#208eb7"</span>, <span class="st">"#6ce9d3"</span>, <span class="st">"#1c5e39"</span>, <span class="st">"#8fca40"</span>, <span class="st">"#154975"</span>, <span class="st">"#b1c8eb"</span>))</code></pre></div>
<p><img src="man/figures/unnamed-chunk-6-2.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#you can customise the cluster color bars (provide color_branch values)</span>
<span class="kw"><a href="../reference/plot_CORE.html">plot_CORE</a></span>(CORE_cluster<span class="op">$</span>tree, CORE_cluster<span class="op">$</span>Cluster, <span class="dt">color_branch =</span> <span class="kw">c</span>(<span class="st">"#208eb7"</span>, <span class="st">"#6ce9d3"</span>, <span class="st">"#1c5e39"</span>, <span class="st">"#8fca40"</span>, <span class="st">"#154975"</span>, <span class="st">"#b1c8eb"</span>))</code></pre></div>
</div>
<div id="plot-the-optimal-clustering-result" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-optimal-clustering-result" class="anchor"></a>3.3 Plot the optimal clustering result</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#extract optimal index identified by CORE_scGPS</span>
optimal_index =<span class="st"> </span><span class="kw">which</span>(CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height <span class="op">==</span><span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes)

<span class="co">#plot the optimal result</span>
<span class="kw"><a href="../reference/plot_optimal_CORE.html">plot_optimal_CORE</a></span>(<span class="dt">original_tree=</span> CORE_cluster<span class="op">$</span>tree, <span class="dt">optimal_cluster =</span> <span class="kw">unlist</span>(CORE_cluster<span class="op">$</span>Cluster[optimal_index]), <span class="dt">shift =</span> <span class="op">-</span><span class="dv">100</span>)
<span class="co">#&gt; [1] "Ordering and assigning labels..."</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 128 270  NA</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; [1] 128 270 393</span>
<span class="co">#&gt; [1] "Plotting the colored dendrogram now...."</span></code></pre></div>
<p><img src="man/figures/unnamed-chunk-7-1.png" width="700"></p>
<pre><code>#&gt; [1] "Plotting the bar underneath now...."</code></pre>
</div>
<div id="compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr" class="anchor"></a>3.4 Compare clustering results with other dimensional reduction methods (e.g., CIDR)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(cidr)
t &lt;-<span class="st"> </span><span class="kw"><a href="../reference/CIDR_scGPS.html">CIDR_scGPS</a></span>(<span class="dt">expression.matrix=</span><span class="kw">assay</span>(mixedpop2))
<span class="co">#&gt; [1] "building cidr object..."</span>
<span class="co">#&gt; [1] "determine dropout candidates..."</span>
<span class="co">#&gt; [1] "determine the imputation weighting threshold..."</span>
<span class="co">#&gt; [1] "computes the _CIDR_ dissimilarity matrix..."</span>
<span class="co">#&gt; [1] "PCA plot with proportion of variance explained..."</span></code></pre></div>
<p><img src="man/figures/unnamed-chunk-8-1.png" width="700"></p>
<pre><code>#&gt; [1] "find the number of PC..."
#&gt; [1] "perform clustering..."
p2 &lt;-plotReduced_scGPS(t, color_fac = factor(colData(mixedpop2)[,1]),palletes =1:length(unique(colData(mixedpop2)[,1])))
#&gt; Warning: package 'cowplot' was built under R version 3.4.3
#&gt; 
#&gt; Attaching package: 'cowplot'
#&gt; The following object is masked from 'package:ggplot2':
#&gt; 
#&gt;     ggsave
p2</code></pre>
<p><img src="man/figures/unnamed-chunk-8-2.png" width="700"></p>
</div>
<div id="find-gene-markers-and-annotate-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#find-gene-markers-and-annotate-clusters" class="anchor"></a>3.5 Find gene markers and annotate clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#load gene list (this can be any lists of user-selected genes)</span>
genes &lt;-GeneList
genes &lt;-genes<span class="op">$</span>Merged_unique

<span class="co">#the gene list can also be objectively identified by differential expression analysis</span>
<span class="co">#cluster information is requied for findMarkers_scGPS. Here, we use CORE results. </span>

Optimal_index &lt;-<span class="st"> </span><span class="kw">which</span>( CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>KeyStats<span class="op">$</span>Height <span class="op">==</span><span class="st"> </span>CORE_cluster<span class="op">$</span>optimalClust<span class="op">$</span>OptimalRes)
<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="kw">unlist</span>(CORE_cluster<span class="op">$</span>Cluster[[Optimal_index]])

<span class="kw">suppressMessages</span>(<span class="kw">library</span>(locfit))
<span class="kw">suppressMessages</span>(<span class="kw">library</span>(DESeq))

DEgenes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/findMarkers_scGPS.html">findMarkers_scGPS</a></span>(<span class="dt">expression_matrix=</span><span class="kw">assay</span>(mixedpop2), <span class="dt">cluster =</span> <span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>],
                             <span class="dt">selected_cluster=</span><span class="kw">unique</span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))
<span class="co">#&gt; [1] "Start estimate dispersions for cluster 1..."</span>
<span class="co">#&gt; [1] "Done estimate dispersions. Start nbinom test for cluster 1..."</span>
<span class="co">#&gt; [1] "Done nbinom test for cluster 1 ..."</span>
<span class="co">#&gt; [1] "Adjust foldchange by subtracting basemean to 1..."</span>
<span class="co">#&gt; [1] "Start estimate dispersions for cluster 2..."</span>
<span class="co">#&gt; [1] "Done estimate dispersions. Start nbinom test for cluster 2..."</span>
<span class="co">#&gt; [1] "Done nbinom test for cluster 2 ..."</span>
<span class="co">#&gt; [1] "Adjust foldchange by subtracting basemean to 1..."</span>
<span class="co">#&gt; [1] "Start estimate dispersions for cluster 3..."</span>
<span class="co">#&gt; [1] "Done estimate dispersions. Start nbinom test for cluster 3..."</span>
<span class="co">#&gt; [1] "Done nbinom test for cluster 3 ..."</span>
<span class="co">#&gt; [1] "Adjust foldchange by subtracting basemean to 1..."</span>

<span class="co">#the output contains dataframes for each cluster.</span>
<span class="co">#the data frame contains all genes, sorted by p-values </span>
<span class="kw">names</span>(DEgenes)
<span class="co">#&gt; [1] "DE_Subpop1vsRemaining" "DE_Subpop2vsRemaining" "DE_Subpop3vsRemaining"</span>

<span class="co">#you can annotate the identified clusters </span>
DEgeneList_3vsOthers &lt;-<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id

<span class="co">#users need to check the format of the gene input to make sure they are consistent to </span>
<span class="co">#the gene names in the expression matrix </span>
DEgeneList_3vsOthers &lt;-<span class="kw">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, DEgeneList_3vsOthers )

<span class="co">#the following command saves the file "PathwayEnrichment.xlsx" to the working dir</span>
<span class="co">#use 500 top DE genes </span>
<span class="kw">suppressMessages</span>(<span class="kw">library</span>(DOSE))
<span class="kw">suppressMessages</span>(<span class="kw">library</span>(ReactomePA))
<span class="kw">suppressMessages</span>(<span class="kw">library</span>(clusterProfiler))

enrichment_test &lt;-<span class="st"> </span><span class="kw"><a href="../reference/annotate_scGPS.html">annotate_scGPS</a></span>(DEgeneList_3vsOthers[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>], <span class="dt">pvalueCutoff=</span><span class="fl">0.05</span>, <span class="dt">gene_symbol=</span><span class="ot">TRUE</span>,<span class="dt">output_filename =</span> <span class="st">"PathwayEnrichment.xlsx"</span>, <span class="dt">output_path =</span> <span class="ot">NULL</span> )
<span class="co">#&gt; [1] "Original gene number in geneList"</span>
<span class="co">#&gt; [1] 500</span>
<span class="co">#&gt; [1] "Number of genes successfully converted"</span>
<span class="co">#&gt; [1] 490</span>

<span class="co">#the enrichment outputs can be displayed by running</span>
<span class="kw"><a href="http://www.rdocumentation.org/packages/DOSE/topics/dotplot-methods">dotplot</a></span>(enrichment_test, <span class="dt">showCategory=</span><span class="dv">15</span>)</code></pre></div>
<p><img src="man/figures/unnamed-chunk-9-1.png" width="700"></p>
</div>
</div>
<div id="relationship-between-clusters-within-one-sample-or-between-two-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples" class="anchor"></a>4. Relationship between clusters within one sample or between two samples</h1>
<p><em>The purpose of this workflow is to solve the following task: given one or two unknown mixed population(s) and clusters in each mixed population, estimate and visualise relationship between clusters</em></p>
<div id="start-the-scgps-prediction-to-find-relationship-between-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#start-the-scgps-prediction-to-find-relationship-between-clusters" class="anchor"></a>4.1 Start the scGPS prediction to find relationship between clusters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#select a subpopulation, and input gene list </span>
c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>]
<span class="co">#format gene names </span>
genes &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, genes)

<span class="co">#run the test bootstrap with nboots = 2 runs</span>
<span class="kw">sink</span>(<span class="st">"temp"</span>)
LSOLDA_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_scGPS.html">bootstrap_scGPS</a></span>(<span class="dt">nboots =</span> <span class="dv">2</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>())
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = t(predictor_S1), y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 -1.922e-15 0.350700</span>
<span class="co">#&gt;   [2,]  1  3.165e-02 0.334800</span>
<span class="co">#&gt;   [3,]  1  6.113e-02 0.319500</span>
<span class="co">#&gt;   [4,]  1  8.909e-02 0.305000</span>
<span class="co">#&gt;   [5,]  1  1.160e-01 0.291200</span>
<span class="co">#&gt;   [6,]  1  1.420e-01 0.277900</span>
<span class="co">#&gt;   [7,]  1  1.672e-01 0.265300</span>
<span class="co">#&gt;   [8,]  1  1.919e-01 0.253200</span>
<span class="co">#&gt;   [9,]  1  2.159e-01 0.241700</span>
<span class="co">#&gt;  [10,]  1  2.392e-01 0.230700</span>
<span class="co">#&gt;  [11,]  1  2.619e-01 0.220200</span>
<span class="co">#&gt;  [12,]  1  2.839e-01 0.210200</span>
<span class="co">#&gt;  [13,]  1  3.053e-01 0.200700</span>
<span class="co">#&gt;  [14,]  1  3.260e-01 0.191600</span>
<span class="co">#&gt;  [15,]  1  3.461e-01 0.182900</span>
<span class="co">#&gt;  [16,]  2  3.658e-01 0.174500</span>
<span class="co">#&gt;  [17,]  2  3.855e-01 0.166600</span>
<span class="co">#&gt;  [18,]  3  4.046e-01 0.159000</span>
<span class="co">#&gt;  [19,]  3  4.232e-01 0.151800</span>
<span class="co">#&gt;  [20,]  3  4.409e-01 0.144900</span>
<span class="co">#&gt;  [21,]  4  4.581e-01 0.138300</span>
<span class="co">#&gt;  [22,]  4  4.747e-01 0.132000</span>
<span class="co">#&gt;  [23,]  4  4.906e-01 0.126000</span>
<span class="co">#&gt;  [24,]  4  5.058e-01 0.120300</span>
<span class="co">#&gt;  [25,]  4  5.204e-01 0.114800</span>
<span class="co">#&gt;  [26,]  4  5.345e-01 0.109600</span>
<span class="co">#&gt;  [27,]  4  5.480e-01 0.104600</span>
<span class="co">#&gt;  [28,]  4  5.609e-01 0.099880</span>
<span class="co">#&gt;  [29,]  5  5.736e-01 0.095340</span>
<span class="co">#&gt;  [30,]  5  5.861e-01 0.091010</span>
<span class="co">#&gt;  [31,]  5  5.981e-01 0.086870</span>
<span class="co">#&gt;  [32,]  6  6.098e-01 0.082920</span>
<span class="co">#&gt;  [33,]  6  6.211e-01 0.079150</span>
<span class="co">#&gt;  [34,]  6  6.320e-01 0.075560</span>
<span class="co">#&gt;  [35,]  6  6.424e-01 0.072120</span>
<span class="co">#&gt;  [36,]  6  6.524e-01 0.068840</span>
<span class="co">#&gt;  [37,]  6  6.622e-01 0.065710</span>
<span class="co">#&gt;  [38,]  6  6.718e-01 0.062730</span>
<span class="co">#&gt;  [39,]  6  6.809e-01 0.059880</span>
<span class="co">#&gt;  [40,]  7  6.898e-01 0.057160</span>
<span class="co">#&gt;  [41,]  8  6.983e-01 0.054560</span>
<span class="co">#&gt;  [42,]  8  7.066e-01 0.052080</span>
<span class="co">#&gt;  [43,]  9  7.146e-01 0.049710</span>
<span class="co">#&gt;  [44,]  9  7.224e-01 0.047450</span>
<span class="co">#&gt;  [45,]  9  7.299e-01 0.045290</span>
<span class="co">#&gt;  [46,]  9  7.373e-01 0.043240</span>
<span class="co">#&gt;  [47,]  9  7.446e-01 0.041270</span>
<span class="co">#&gt;  [48,]  9  7.516e-01 0.039390</span>
<span class="co">#&gt;  [49,]  9  7.583e-01 0.037600</span>
<span class="co">#&gt;  [50,]  9  7.648e-01 0.035900</span>
<span class="co">#&gt;  [51,] 10  7.711e-01 0.034260</span>
<span class="co">#&gt;  [52,] 11  7.775e-01 0.032710</span>
<span class="co">#&gt;  [53,] 13  7.840e-01 0.031220</span>
<span class="co">#&gt;  [54,] 13  7.903e-01 0.029800</span>
<span class="co">#&gt;  [55,] 16  7.966e-01 0.028450</span>
<span class="co">#&gt;  [56,] 16  8.026e-01 0.027150</span>
<span class="co">#&gt;  [57,] 18  8.086e-01 0.025920</span>
<span class="co">#&gt;  [58,] 19  8.145e-01 0.024740</span>
<span class="co">#&gt;  [59,] 20  8.203e-01 0.023620</span>
<span class="co">#&gt;  [60,] 22  8.259e-01 0.022540</span>
<span class="co">#&gt;  [61,] 23  8.315e-01 0.021520</span>
<span class="co">#&gt;  [62,] 24  8.369e-01 0.020540</span>
<span class="co">#&gt;  [63,] 24  8.422e-01 0.019610</span>
<span class="co">#&gt;  [64,] 26  8.473e-01 0.018720</span>
<span class="co">#&gt;  [65,] 28  8.527e-01 0.017870</span>
<span class="co">#&gt;  [66,] 30  8.583e-01 0.017050</span>
<span class="co">#&gt;  [67,] 31  8.636e-01 0.016280</span>
<span class="co">#&gt;  [68,] 32  8.687e-01 0.015540</span>
<span class="co">#&gt;  [69,] 35  8.738e-01 0.014830</span>
<span class="co">#&gt;  [70,] 36  8.787e-01 0.014160</span>
<span class="co">#&gt;  [71,] 39  8.835e-01 0.013510</span>
<span class="co">#&gt;  [72,] 38  8.883e-01 0.012900</span>
<span class="co">#&gt;  [73,] 36  8.929e-01 0.012310</span>
<span class="co">#&gt;  [74,] 36  8.973e-01 0.011750</span>
<span class="co">#&gt;  [75,] 36  9.015e-01 0.011220</span>
<span class="co">#&gt;  [76,] 37  9.055e-01 0.010710</span>
<span class="co">#&gt;  [77,] 37  9.094e-01 0.010220</span>
<span class="co">#&gt;  [78,] 39  9.131e-01 0.009758</span>
<span class="co">#&gt;  [79,] 42  9.168e-01 0.009315</span>
<span class="co">#&gt;  [80,] 44  9.204e-01 0.008892</span>
<span class="co">#&gt;  [81,] 44  9.238e-01 0.008487</span>
<span class="co">#&gt;  [82,] 45  9.271e-01 0.008102</span>
<span class="co">#&gt;  [83,] 46  9.302e-01 0.007733</span>
<span class="co">#&gt;  [84,] 47  9.333e-01 0.007382</span>
<span class="co">#&gt;  [85,] 47  9.362e-01 0.007046</span>
<span class="co">#&gt;  [86,] 49  9.389e-01 0.006726</span>
<span class="co">#&gt;  [87,] 49  9.416e-01 0.006420</span>
<span class="co">#&gt;  [88,] 50  9.442e-01 0.006129</span>
<span class="co">#&gt;  [89,] 51  9.466e-01 0.005850</span>
<span class="co">#&gt;  [90,] 50  9.490e-01 0.005584</span>
<span class="co">#&gt;  [91,] 52  9.512e-01 0.005330</span>
<span class="co">#&gt;  [92,] 55  9.534e-01 0.005088</span>
<span class="co">#&gt;  [93,] 56  9.555e-01 0.004857</span>
<span class="co">#&gt;  [94,] 56  9.575e-01 0.004636</span>
<span class="co">#&gt;  [95,] 56  9.594e-01 0.004425</span>
<span class="co">#&gt;  [96,] 56  9.612e-01 0.004224</span>
<span class="co">#&gt;  [97,] 56  9.630e-01 0.004032</span>
<span class="co">#&gt;  [98,] 55  9.646e-01 0.003849</span>
<span class="co">#&gt;  [99,] 55  9.662e-01 0.003674</span>
<span class="co">#&gt; [100,] 56  9.678e-01 0.003507</span>
<span class="co">#&gt; [1] "done bootstrap 1"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = t(predictor_S1), y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df       %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 -1.922e-15 0.366400</span>
<span class="co">#&gt;   [2,]  1  3.449e-02 0.349800</span>
<span class="co">#&gt;   [3,]  1  6.631e-02 0.333900</span>
<span class="co">#&gt;   [4,]  1  9.600e-02 0.318700</span>
<span class="co">#&gt;   [5,]  1  1.239e-01 0.304200</span>
<span class="co">#&gt;   [6,]  1  1.505e-01 0.290400</span>
<span class="co">#&gt;   [7,]  1  1.758e-01 0.277200</span>
<span class="co">#&gt;   [8,]  1  2.000e-01 0.264600</span>
<span class="co">#&gt;   [9,]  1  2.234e-01 0.252600</span>
<span class="co">#&gt;  [10,]  1  2.459e-01 0.241100</span>
<span class="co">#&gt;  [11,]  1  2.677e-01 0.230100</span>
<span class="co">#&gt;  [12,]  1  2.889e-01 0.219700</span>
<span class="co">#&gt;  [13,]  1  3.095e-01 0.209700</span>
<span class="co">#&gt;  [14,]  1  3.294e-01 0.200100</span>
<span class="co">#&gt;  [15,]  1  3.488e-01 0.191000</span>
<span class="co">#&gt;  [16,]  1  3.677e-01 0.182400</span>
<span class="co">#&gt;  [17,]  1  3.860e-01 0.174100</span>
<span class="co">#&gt;  [18,]  1  4.038e-01 0.166200</span>
<span class="co">#&gt;  [19,]  1  4.211e-01 0.158600</span>
<span class="co">#&gt;  [20,]  1  4.378e-01 0.151400</span>
<span class="co">#&gt;  [21,]  2  4.543e-01 0.144500</span>
<span class="co">#&gt;  [22,]  2  4.713e-01 0.138000</span>
<span class="co">#&gt;  [23,]  3  4.876e-01 0.131700</span>
<span class="co">#&gt;  [24,]  3  5.038e-01 0.125700</span>
<span class="co">#&gt;  [25,]  3  5.192e-01 0.120000</span>
<span class="co">#&gt;  [26,]  3  5.341e-01 0.114500</span>
<span class="co">#&gt;  [27,]  3  5.484e-01 0.109300</span>
<span class="co">#&gt;  [28,]  3  5.622e-01 0.104400</span>
<span class="co">#&gt;  [29,]  3  5.754e-01 0.099610</span>
<span class="co">#&gt;  [30,]  3  5.881e-01 0.095080</span>
<span class="co">#&gt;  [31,]  3  6.003e-01 0.090760</span>
<span class="co">#&gt;  [32,]  3  6.121e-01 0.086640</span>
<span class="co">#&gt;  [33,]  3  6.234e-01 0.082700</span>
<span class="co">#&gt;  [34,]  4  6.343e-01 0.078940</span>
<span class="co">#&gt;  [35,]  4  6.448e-01 0.075350</span>
<span class="co">#&gt;  [36,]  4  6.550e-01 0.071930</span>
<span class="co">#&gt;  [37,]  4  6.647e-01 0.068660</span>
<span class="co">#&gt;  [38,]  6  6.741e-01 0.065540</span>
<span class="co">#&gt;  [39,]  6  6.834e-01 0.062560</span>
<span class="co">#&gt;  [40,]  6  6.923e-01 0.059720</span>
<span class="co">#&gt;  [41,]  6  7.009e-01 0.057000</span>
<span class="co">#&gt;  [42,]  7  7.094e-01 0.054410</span>
<span class="co">#&gt;  [43,]  8  7.177e-01 0.051940</span>
<span class="co">#&gt;  [44,]  8  7.257e-01 0.049580</span>
<span class="co">#&gt;  [45,]  8  7.333e-01 0.047320</span>
<span class="co">#&gt;  [46,]  8  7.407e-01 0.045170</span>
<span class="co">#&gt;  [47,] 11  7.478e-01 0.043120</span>
<span class="co">#&gt;  [48,] 11  7.548e-01 0.041160</span>
<span class="co">#&gt;  [49,] 12  7.614e-01 0.039290</span>
<span class="co">#&gt;  [50,] 13  7.679e-01 0.037500</span>
<span class="co">#&gt;  [51,] 14  7.742e-01 0.035800</span>
<span class="co">#&gt;  [52,] 15  7.804e-01 0.034170</span>
<span class="co">#&gt;  [53,] 15  7.865e-01 0.032620</span>
<span class="co">#&gt;  [54,] 15  7.923e-01 0.031140</span>
<span class="co">#&gt;  [55,] 17  7.981e-01 0.029720</span>
<span class="co">#&gt;  [56,] 18  8.039e-01 0.028370</span>
<span class="co">#&gt;  [57,] 22  8.097e-01 0.027080</span>
<span class="co">#&gt;  [58,] 22  8.154e-01 0.025850</span>
<span class="co">#&gt;  [59,] 23  8.209e-01 0.024670</span>
<span class="co">#&gt;  [60,] 24  8.264e-01 0.023550</span>
<span class="co">#&gt;  [61,] 24  8.318e-01 0.022480</span>
<span class="co">#&gt;  [62,] 25  8.371e-01 0.021460</span>
<span class="co">#&gt;  [63,] 27  8.422e-01 0.020490</span>
<span class="co">#&gt;  [64,] 27  8.473e-01 0.019550</span>
<span class="co">#&gt;  [65,] 27  8.524e-01 0.018670</span>
<span class="co">#&gt;  [66,] 28  8.574e-01 0.017820</span>
<span class="co">#&gt;  [67,] 30  8.623e-01 0.017010</span>
<span class="co">#&gt;  [68,] 32  8.671e-01 0.016230</span>
<span class="co">#&gt;  [69,] 34  8.720e-01 0.015500</span>
<span class="co">#&gt;  [70,] 35  8.769e-01 0.014790</span>
<span class="co">#&gt;  [71,] 38  8.819e-01 0.014120</span>
<span class="co">#&gt;  [72,] 40  8.868e-01 0.013480</span>
<span class="co">#&gt;  [73,] 42  8.915e-01 0.012870</span>
<span class="co">#&gt;  [74,] 43  8.961e-01 0.012280</span>
<span class="co">#&gt;  [75,] 45  9.006e-01 0.011720</span>
<span class="co">#&gt;  [76,] 47  9.050e-01 0.011190</span>
<span class="co">#&gt;  [77,] 48  9.094e-01 0.010680</span>
<span class="co">#&gt;  [78,] 49  9.135e-01 0.010200</span>
<span class="co">#&gt;  [79,] 50  9.175e-01 0.009732</span>
<span class="co">#&gt;  [80,] 50  9.213e-01 0.009290</span>
<span class="co">#&gt;  [81,] 50  9.249e-01 0.008868</span>
<span class="co">#&gt;  [82,] 52  9.284e-01 0.008465</span>
<span class="co">#&gt;  [83,] 51  9.316e-01 0.008080</span>
<span class="co">#&gt;  [84,] 51  9.347e-01 0.007713</span>
<span class="co">#&gt;  [85,] 51  9.377e-01 0.007362</span>
<span class="co">#&gt;  [86,] 52  9.405e-01 0.007027</span>
<span class="co">#&gt;  [87,] 54  9.432e-01 0.006708</span>
<span class="co">#&gt;  [88,] 56  9.458e-01 0.006403</span>
<span class="co">#&gt;  [89,] 56  9.483e-01 0.006112</span>
<span class="co">#&gt;  [90,] 56  9.506e-01 0.005834</span>
<span class="co">#&gt;  [91,] 56  9.529e-01 0.005569</span>
<span class="co">#&gt;  [92,] 54  9.550e-01 0.005316</span>
<span class="co">#&gt;  [93,] 54  9.571e-01 0.005074</span>
<span class="co">#&gt;  [94,] 55  9.590e-01 0.004844</span>
<span class="co">#&gt;  [95,] 55  9.609e-01 0.004624</span>
<span class="co">#&gt;  [96,] 56  9.626e-01 0.004413</span>
<span class="co">#&gt;  [97,] 56  9.643e-01 0.004213</span>
<span class="co">#&gt;  [98,] 57  9.660e-01 0.004021</span>
<span class="co">#&gt;  [99,] 57  9.675e-01 0.003839</span>
<span class="co">#&gt; [100,] 58  9.690e-01 0.003664</span>
<span class="co">#&gt; [1] "done bootstrap 2"</span>
<span class="kw">sink</span>()</code></pre></div>
</div>
<div id="display-summary-results-for-the-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#display-summary-results-for-the-prediction" class="anchor"></a>4.2 Display summary results for the prediction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#get the number of rows for the summary matrix </span>
row_cluster &lt;-<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">colData</span>(mixedpop2)[,<span class="dv">1</span>]))

<span class="co">#summary results LDA</span>
<span class="kw"><a href="../reference/summary_prediction_lda.html">summary_prediction_lda</a></span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster )
<span class="co">#&gt;                 V1               V2                                names</span>
<span class="co">#&gt; 1         85.15625         86.71875 LDA for subpop 1 in target mixedpop2</span>
<span class="co">#&gt; 2  13.953488372093 13.4883720930233 LDA for subpop 2 in target mixedpop2</span>
<span class="co">#&gt; 3 6.89655172413793                0 LDA for subpop 3 in target mixedpop2</span>

<span class="co">#summary results Lasso</span>
<span class="kw"><a href="../reference/summary_prediction_lasso.html">summary_prediction_lasso</a></span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> row_cluster)
<span class="co">#&gt;                 V1               V2                                 names</span>
<span class="co">#&gt; 1         99.21875        97.265625 LASSO for subpop1 in target mixedpop2</span>
<span class="co">#&gt; 2 4.18604651162791 4.18604651162791 LASSO for subpop2 in target mixedpop2</span>
<span class="co">#&gt; 3               NA               NA LASSO for subpop3 in target mixedpop2</span>

<span class="co">#summary deviance </span>
<span class="kw"><a href="../reference/summary_deviance.html">summary_deviance</a></span>(LSOLDA_dat)
<span class="co">#&gt; $allDeviance</span>
<span class="co">#&gt; [1] "0.6809" "0.6647"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $DeviMax</span>
<span class="co">#&gt;         Dfd   Deviance        DEgenes</span>
<span class="co">#&gt; 1         0 -1.922e-15 genes_cluster1</span>
<span class="co">#&gt; 2         1     0.3461 genes_cluster1</span>
<span class="co">#&gt; 3         2     0.3855 genes_cluster1</span>
<span class="co">#&gt; 4         3     0.4409 genes_cluster1</span>
<span class="co">#&gt; 5         4     0.5609 genes_cluster1</span>
<span class="co">#&gt; 6         5     0.5981 genes_cluster1</span>
<span class="co">#&gt; 7         6     0.6809 genes_cluster1</span>
<span class="co">#&gt; 8 remaining          1        DEgenes</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $LassoGenesMax</span>
<span class="co">#&gt;                                       1                       name</span>
<span class="co">#&gt; (Intercept)                -3.635155864                (Intercept)</span>
<span class="co">#&gt; PPIC_ENSG00000168938        0.047213016       PPIC_ENSG00000168938</span>
<span class="co">#&gt; AC002454.1_ENSG00000237819  0.020072807 AC002454.1_ENSG00000237819</span>
<span class="co">#&gt; MDK_ENSG00000110492         0.004522720        MDK_ENSG00000110492</span>
<span class="co">#&gt; MALAT1_ENSG00000251562      0.011222896     MALAT1_ENSG00000251562</span>
<span class="co">#&gt; NPC2_ENSG00000119655        0.013680130       NPC2_ENSG00000119655</span>
<span class="co">#&gt; TPM1_ENSG00000140416        0.004579622       TPM1_ENSG00000140416</span></code></pre></div>
</div>
<div id="plot-the-relationship-between-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-relationship-between-clusters" class="anchor"></a>4.3 Plot the relationship between clusters</h2>
<p><em>Here we look at one example use case to find relationship between clusters within one sample or between two sample</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run prediction for 3 clusters </span>

c_selectID &lt;-<span class="st"> </span><span class="dv">1</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop1vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] <span class="co">#top 200 gene markers distinguishing cluster 1 </span>
genes &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, genes)

LSOLDA_dat1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_scGPS.html">bootstrap_scGPS</a></span>(<span class="dt">nboots =</span> <span class="dv">1</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>())
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = t(predictor_S1), y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;       Df       %Dev    Lambda</span>
<span class="co">#&gt;  [1,]  0 -1.922e-15 3.472e-01</span>
<span class="co">#&gt;  [2,]  1  5.989e-02 3.164e-01</span>
<span class="co">#&gt;  [3,]  1  1.135e-01 2.882e-01</span>
<span class="co">#&gt;  [4,]  1  1.635e-01 2.626e-01</span>
<span class="co">#&gt;  [5,]  1  2.109e-01 2.393e-01</span>
<span class="co">#&gt;  [6,]  1  2.559e-01 2.180e-01</span>
<span class="co">#&gt;  [7,]  1  2.984e-01 1.987e-01</span>
<span class="co">#&gt;  [8,]  1  3.386e-01 1.810e-01</span>
<span class="co">#&gt;  [9,]  2  3.776e-01 1.649e-01</span>
<span class="co">#&gt; [10,]  2  4.145e-01 1.503e-01</span>
<span class="co">#&gt; [11,]  4  4.495e-01 1.369e-01</span>
<span class="co">#&gt; [12,]  5  4.834e-01 1.248e-01</span>
<span class="co">#&gt; [13,]  7  5.151e-01 1.137e-01</span>
<span class="co">#&gt; [14,]  8  5.449e-01 1.036e-01</span>
<span class="co">#&gt; [15,]  8  5.725e-01 9.439e-02</span>
<span class="co">#&gt; [16,]  7  5.978e-01 8.600e-02</span>
<span class="co">#&gt; [17,]  7  6.211e-01 7.836e-02</span>
<span class="co">#&gt; [18,]  9  6.431e-01 7.140e-02</span>
<span class="co">#&gt; [19,] 10  6.639e-01 6.506e-02</span>
<span class="co">#&gt; [20,] 10  6.834e-01 5.928e-02</span>
<span class="co">#&gt; [21,]  9  7.013e-01 5.401e-02</span>
<span class="co">#&gt; [22,] 12  7.181e-01 4.921e-02</span>
<span class="co">#&gt; [23,] 12  7.338e-01 4.484e-02</span>
<span class="co">#&gt; [24,] 13  7.487e-01 4.086e-02</span>
<span class="co">#&gt; [25,] 13  7.626e-01 3.723e-02</span>
<span class="co">#&gt; [26,] 14  7.757e-01 3.392e-02</span>
<span class="co">#&gt; [27,] 13  7.885e-01 3.091e-02</span>
<span class="co">#&gt; [28,] 13  8.004e-01 2.816e-02</span>
<span class="co">#&gt; [29,] 13  8.115e-01 2.566e-02</span>
<span class="co">#&gt; [30,] 14  8.219e-01 2.338e-02</span>
<span class="co">#&gt; [31,] 15  8.321e-01 2.130e-02</span>
<span class="co">#&gt; [32,] 14  8.415e-01 1.941e-02</span>
<span class="co">#&gt; [33,] 15  8.503e-01 1.769e-02</span>
<span class="co">#&gt; [34,] 16  8.586e-01 1.612e-02</span>
<span class="co">#&gt; [35,] 17  8.667e-01 1.468e-02</span>
<span class="co">#&gt; [36,] 18  8.748e-01 1.338e-02</span>
<span class="co">#&gt; [37,] 19  8.827e-01 1.219e-02</span>
<span class="co">#&gt; [38,] 21  8.903e-01 1.111e-02</span>
<span class="co">#&gt; [39,] 23  8.977e-01 1.012e-02</span>
<span class="co">#&gt; [40,] 24  9.046e-01 9.222e-03</span>
<span class="co">#&gt; [41,] 25  9.112e-01 8.403e-03</span>
<span class="co">#&gt; [42,] 27  9.178e-01 7.656e-03</span>
<span class="co">#&gt; [43,] 28  9.242e-01 6.976e-03</span>
<span class="co">#&gt; [44,] 30  9.304e-01 6.356e-03</span>
<span class="co">#&gt; [45,] 30  9.362e-01 5.792e-03</span>
<span class="co">#&gt; [46,] 30  9.415e-01 5.277e-03</span>
<span class="co">#&gt; [47,] 33  9.464e-01 4.808e-03</span>
<span class="co">#&gt; [48,] 36  9.510e-01 4.381e-03</span>
<span class="co">#&gt; [49,] 36  9.553e-01 3.992e-03</span>
<span class="co">#&gt; [50,] 37  9.592e-01 3.637e-03</span>
<span class="co">#&gt; [51,] 38  9.628e-01 3.314e-03</span>
<span class="co">#&gt; [52,] 38  9.660e-01 3.020e-03</span>
<span class="co">#&gt; [53,] 39  9.690e-01 2.751e-03</span>
<span class="co">#&gt; [54,] 39  9.717e-01 2.507e-03</span>
<span class="co">#&gt; [55,] 40  9.742e-01 2.284e-03</span>
<span class="co">#&gt; [56,] 40  9.765e-01 2.081e-03</span>
<span class="co">#&gt; [57,] 40  9.785e-01 1.896e-03</span>
<span class="co">#&gt; [58,] 40  9.804e-01 1.728e-03</span>
<span class="co">#&gt; [59,] 41  9.821e-01 1.574e-03</span>
<span class="co">#&gt; [60,] 43  9.837e-01 1.435e-03</span>
<span class="co">#&gt; [61,] 43  9.851e-01 1.307e-03</span>
<span class="co">#&gt; [62,] 43  9.864e-01 1.191e-03</span>
<span class="co">#&gt; [63,] 43  9.876e-01 1.085e-03</span>
<span class="co">#&gt; [64,] 43  9.887e-01 9.888e-04</span>
<span class="co">#&gt; [65,] 44  9.897e-01 9.010e-04</span>
<span class="co">#&gt; [66,] 44  9.906e-01 8.209e-04</span>
<span class="co">#&gt; [67,] 44  9.914e-01 7.480e-04</span>
<span class="co">#&gt; [68,] 43  9.922e-01 6.816e-04</span>
<span class="co">#&gt; [69,] 43  9.929e-01 6.210e-04</span>
<span class="co">#&gt; [70,] 43  9.935e-01 5.658e-04</span>
<span class="co">#&gt; [71,] 43  9.941e-01 5.156e-04</span>
<span class="co">#&gt; [72,] 43  9.946e-01 4.698e-04</span>
<span class="co">#&gt; [73,] 44  9.951e-01 4.280e-04</span>
<span class="co">#&gt; [74,] 43  9.955e-01 3.900e-04</span>
<span class="co">#&gt; [75,] 43  9.959e-01 3.554e-04</span>
<span class="co">#&gt; [76,] 44  9.963e-01 3.238e-04</span>
<span class="co">#&gt; [77,] 44  9.966e-01 2.950e-04</span>
<span class="co">#&gt; [78,] 43  9.969e-01 2.688e-04</span>
<span class="co">#&gt; [79,] 43  9.972e-01 2.449e-04</span>
<span class="co">#&gt; [80,] 42  9.974e-01 2.232e-04</span>
<span class="co">#&gt; [81,] 42  9.977e-01 2.034e-04</span>
<span class="co">#&gt; [82,] 42  9.979e-01 1.853e-04</span>
<span class="co">#&gt; [83,] 41  9.980e-01 1.688e-04</span>
<span class="co">#&gt; [84,] 41  9.982e-01 1.538e-04</span>
<span class="co">#&gt; [85,] 41  9.984e-01 1.402e-04</span>
<span class="co">#&gt; [86,] 41  9.985e-01 1.277e-04</span>
<span class="co">#&gt; [87,] 41  9.987e-01 1.164e-04</span>
<span class="co">#&gt; [88,] 41  9.988e-01 1.060e-04</span>
<span class="co">#&gt; [89,] 41  9.989e-01 9.661e-05</span>
<span class="co">#&gt; [90,] 43  9.990e-01 8.803e-05</span>
<span class="co">#&gt; [91,] 44  9.991e-01 8.021e-05</span>
<span class="co">#&gt; [1] "done bootstrap 1"</span>


c_selectID &lt;-<span class="st"> </span><span class="dv">2</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop2vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
genes &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, genes)
LSOLDA_dat2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_scGPS.html">bootstrap_scGPS</a></span>(<span class="dt">nboots =</span> <span class="dv">1</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>())
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = t(predictor_S1), y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;       Df       %Dev    Lambda</span>
<span class="co">#&gt;  [1,]  0 -2.403e-15 2.468e-01</span>
<span class="co">#&gt;  [2,]  2  3.250e-02 2.249e-01</span>
<span class="co">#&gt;  [3,]  3  7.067e-02 2.049e-01</span>
<span class="co">#&gt;  [4,]  7  1.075e-01 1.867e-01</span>
<span class="co">#&gt;  [5,]  8  1.435e-01 1.701e-01</span>
<span class="co">#&gt;  [6,] 11  1.787e-01 1.550e-01</span>
<span class="co">#&gt;  [7,] 11  2.101e-01 1.412e-01</span>
<span class="co">#&gt;  [8,] 13  2.375e-01 1.287e-01</span>
<span class="co">#&gt;  [9,] 16  2.655e-01 1.173e-01</span>
<span class="co">#&gt; [10,] 16  2.905e-01 1.068e-01</span>
<span class="co">#&gt; [11,] 19  3.145e-01 9.735e-02</span>
<span class="co">#&gt; [12,] 20  3.372e-01 8.870e-02</span>
<span class="co">#&gt; [13,] 22  3.605e-01 8.082e-02</span>
<span class="co">#&gt; [14,] 26  3.838e-01 7.364e-02</span>
<span class="co">#&gt; [15,] 27  4.077e-01 6.710e-02</span>
<span class="co">#&gt; [16,] 31  4.306e-01 6.114e-02</span>
<span class="co">#&gt; [17,] 31  4.536e-01 5.571e-02</span>
<span class="co">#&gt; [18,] 33  4.754e-01 5.076e-02</span>
<span class="co">#&gt; [19,] 34  4.966e-01 4.625e-02</span>
<span class="co">#&gt; [20,] 33  5.157e-01 4.214e-02</span>
<span class="co">#&gt; [21,] 36  5.344e-01 3.840e-02</span>
<span class="co">#&gt; [22,] 40  5.556e-01 3.499e-02</span>
<span class="co">#&gt; [23,] 44  5.770e-01 3.188e-02</span>
<span class="co">#&gt; [24,] 43  5.964e-01 2.905e-02</span>
<span class="co">#&gt; [25,] 44  6.140e-01 2.647e-02</span>
<span class="co">#&gt; [26,] 47  6.314e-01 2.411e-02</span>
<span class="co">#&gt; [27,] 49  6.498e-01 2.197e-02</span>
<span class="co">#&gt; [28,] 49  6.673e-01 2.002e-02</span>
<span class="co">#&gt; [29,] 49  6.847e-01 1.824e-02</span>
<span class="co">#&gt; [30,] 49  7.014e-01 1.662e-02</span>
<span class="co">#&gt; [31,] 51  7.179e-01 1.514e-02</span>
<span class="co">#&gt; [32,] 53  7.344e-01 1.380e-02</span>
<span class="co">#&gt; [33,] 57  7.508e-01 1.257e-02</span>
<span class="co">#&gt; [34,] 59  7.672e-01 1.146e-02</span>
<span class="co">#&gt; [35,] 60  7.830e-01 1.044e-02</span>
<span class="co">#&gt; [36,] 59  7.982e-01 9.511e-03</span>
<span class="co">#&gt; [37,] 59  8.123e-01 8.666e-03</span>
<span class="co">#&gt; [38,] 59  8.258e-01 7.896e-03</span>
<span class="co">#&gt; [39,] 62  8.388e-01 7.195e-03</span>
<span class="co">#&gt; [40,] 63  8.519e-01 6.556e-03</span>
<span class="co">#&gt; [41,] 65  8.643e-01 5.973e-03</span>
<span class="co">#&gt; [42,] 64  8.758e-01 5.443e-03</span>
<span class="co">#&gt; [43,] 65  8.864e-01 4.959e-03</span>
<span class="co">#&gt; [44,] 67  8.963e-01 4.519e-03</span>
<span class="co">#&gt; [45,] 68  9.054e-01 4.117e-03</span>
<span class="co">#&gt; [46,] 69  9.137e-01 3.751e-03</span>
<span class="co">#&gt; [47,] 70  9.213e-01 3.418e-03</span>
<span class="co">#&gt; [48,] 70  9.283e-01 3.114e-03</span>
<span class="co">#&gt; [49,] 69  9.346e-01 2.838e-03</span>
<span class="co">#&gt; [50,] 69  9.404e-01 2.586e-03</span>
<span class="co">#&gt; [51,] 70  9.456e-01 2.356e-03</span>
<span class="co">#&gt; [52,] 70  9.504e-01 2.147e-03</span>
<span class="co">#&gt; [53,] 70  9.548e-01 1.956e-03</span>
<span class="co">#&gt; [54,] 72  9.587e-01 1.782e-03</span>
<span class="co">#&gt; [55,] 74  9.624e-01 1.624e-03</span>
<span class="co">#&gt; [56,] 75  9.658e-01 1.480e-03</span>
<span class="co">#&gt; [57,] 74  9.689e-01 1.348e-03</span>
<span class="co">#&gt; [58,] 75  9.717e-01 1.228e-03</span>
<span class="co">#&gt; [59,] 75  9.742e-01 1.119e-03</span>
<span class="co">#&gt; [60,] 75  9.765e-01 1.020e-03</span>
<span class="co">#&gt; [61,] 75  9.786e-01 9.292e-04</span>
<span class="co">#&gt; [62,] 77  9.805e-01 8.467e-04</span>
<span class="co">#&gt; [63,] 77  9.823e-01 7.715e-04</span>
<span class="co">#&gt; [64,] 78  9.839e-01 7.029e-04</span>
<span class="co">#&gt; [65,] 78  9.853e-01 6.405e-04</span>
<span class="co">#&gt; [66,] 78  9.866e-01 5.836e-04</span>
<span class="co">#&gt; [67,] 79  9.878e-01 5.317e-04</span>
<span class="co">#&gt; [68,] 80  9.889e-01 4.845e-04</span>
<span class="co">#&gt; [69,] 80  9.899e-01 4.415e-04</span>
<span class="co">#&gt; [70,] 80  9.908e-01 4.022e-04</span>
<span class="co">#&gt; [71,] 81  9.916e-01 3.665e-04</span>
<span class="co">#&gt; [72,] 81  9.923e-01 3.340e-04</span>
<span class="co">#&gt; [73,] 81  9.930e-01 3.043e-04</span>
<span class="co">#&gt; [74,] 81  9.936e-01 2.773e-04</span>
<span class="co">#&gt; [75,] 81  9.942e-01 2.526e-04</span>
<span class="co">#&gt; [76,] 82  9.947e-01 2.302e-04</span>
<span class="co">#&gt; [77,] 83  9.952e-01 2.097e-04</span>
<span class="co">#&gt; [78,] 83  9.956e-01 1.911e-04</span>
<span class="co">#&gt; [79,] 83  9.960e-01 1.741e-04</span>
<span class="co">#&gt; [80,] 83  9.964e-01 1.587e-04</span>
<span class="co">#&gt; [81,] 83  9.967e-01 1.446e-04</span>
<span class="co">#&gt; [82,] 83  9.970e-01 1.317e-04</span>
<span class="co">#&gt; [83,] 82  9.972e-01 1.200e-04</span>
<span class="co">#&gt; [84,] 82  9.975e-01 1.094e-04</span>
<span class="co">#&gt; [85,] 82  9.977e-01 9.964e-05</span>
<span class="co">#&gt; [86,] 82  9.979e-01 9.079e-05</span>
<span class="co">#&gt; [87,] 82  9.981e-01 8.272e-05</span>
<span class="co">#&gt; [88,] 82  9.983e-01 7.537e-05</span>
<span class="co">#&gt; [89,] 82  9.984e-01 6.868e-05</span>
<span class="co">#&gt; [90,] 84  9.986e-01 6.258e-05</span>
<span class="co">#&gt; [91,] 85  9.987e-01 5.702e-05</span>
<span class="co">#&gt; [92,] 85  9.988e-01 5.195e-05</span>
<span class="co">#&gt; [93,] 85  9.989e-01 4.734e-05</span>
<span class="co">#&gt; [94,] 85  9.990e-01 4.313e-05</span>
<span class="co">#&gt; [95,] 85  9.991e-01 3.930e-05</span>
<span class="co">#&gt; [1] "done bootstrap 1"</span>


c_selectID &lt;-<span class="st"> </span><span class="dv">3</span>
genes =<span class="st"> </span>DEgenes<span class="op">$</span>DE_Subpop3vsRemaining<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>]
genes &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, genes)
LSOLDA_dat3 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bootstrap_scGPS.html">bootstrap_scGPS</a></span>(<span class="dt">nboots =</span> <span class="dv">1</span>,<span class="dt">mixedpop1 =</span> mixedpop2, <span class="dt">mixedpop2 =</span> mixedpop2, <span class="dt">genes=</span>genes, c_selectID, <span class="dt">listData =</span><span class="kw">list</span>())
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glmnet(x = t(predictor_S1), y = y_cat, family = "binomial") </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        Df    %Dev   Lambda</span>
<span class="co">#&gt;   [1,]  0 0.00000 0.455500</span>
<span class="co">#&gt;   [2,]  1 0.05323 0.434800</span>
<span class="co">#&gt;   [3,]  1 0.10200 0.415000</span>
<span class="co">#&gt;   [4,]  1 0.14680 0.396200</span>
<span class="co">#&gt;   [5,]  1 0.18830 0.378200</span>
<span class="co">#&gt;   [6,]  1 0.22670 0.361000</span>
<span class="co">#&gt;   [7,]  1 0.26240 0.344600</span>
<span class="co">#&gt;   [8,]  1 0.29570 0.328900</span>
<span class="co">#&gt;   [9,]  1 0.32690 0.314000</span>
<span class="co">#&gt;  [10,]  1 0.35610 0.299700</span>
<span class="co">#&gt;  [11,]  1 0.38350 0.286100</span>
<span class="co">#&gt;  [12,]  1 0.40930 0.273100</span>
<span class="co">#&gt;  [13,]  1 0.43360 0.260700</span>
<span class="co">#&gt;  [14,]  1 0.45660 0.248800</span>
<span class="co">#&gt;  [15,]  1 0.47830 0.237500</span>
<span class="co">#&gt;  [16,]  1 0.49890 0.226700</span>
<span class="co">#&gt;  [17,]  1 0.51840 0.216400</span>
<span class="co">#&gt;  [18,]  1 0.53700 0.206600</span>
<span class="co">#&gt;  [19,]  1 0.55460 0.197200</span>
<span class="co">#&gt;  [20,]  1 0.57140 0.188200</span>
<span class="co">#&gt;  [21,]  1 0.58730 0.179700</span>
<span class="co">#&gt;  [22,]  1 0.60260 0.171500</span>
<span class="co">#&gt;  [23,]  2 0.61950 0.163700</span>
<span class="co">#&gt;  [24,]  2 0.63640 0.156300</span>
<span class="co">#&gt;  [25,]  2 0.65240 0.149200</span>
<span class="co">#&gt;  [26,]  2 0.66760 0.142400</span>
<span class="co">#&gt;  [27,]  2 0.68200 0.135900</span>
<span class="co">#&gt;  [28,]  2 0.69580 0.129700</span>
<span class="co">#&gt;  [29,]  2 0.70890 0.123800</span>
<span class="co">#&gt;  [30,]  2 0.72130 0.118200</span>
<span class="co">#&gt;  [31,]  3 0.73380 0.112800</span>
<span class="co">#&gt;  [32,]  3 0.74590 0.107700</span>
<span class="co">#&gt;  [33,]  3 0.75750 0.102800</span>
<span class="co">#&gt;  [34,]  3 0.76850 0.098140</span>
<span class="co">#&gt;  [35,]  3 0.77900 0.093680</span>
<span class="co">#&gt;  [36,]  3 0.78900 0.089420</span>
<span class="co">#&gt;  [37,]  3 0.79850 0.085350</span>
<span class="co">#&gt;  [38,]  3 0.80750 0.081470</span>
<span class="co">#&gt;  [39,]  3 0.81610 0.077770</span>
<span class="co">#&gt;  [40,]  3 0.82440 0.074240</span>
<span class="co">#&gt;  [41,]  3 0.83220 0.070860</span>
<span class="co">#&gt;  [42,]  3 0.83970 0.067640</span>
<span class="co">#&gt;  [43,]  3 0.84680 0.064570</span>
<span class="co">#&gt;  [44,]  3 0.85360 0.061630</span>
<span class="co">#&gt;  [45,]  3 0.86010 0.058830</span>
<span class="co">#&gt;  [46,]  4 0.86630 0.056160</span>
<span class="co">#&gt;  [47,]  4 0.87230 0.053600</span>
<span class="co">#&gt;  [48,]  4 0.87800 0.051170</span>
<span class="co">#&gt;  [49,]  4 0.88350 0.048840</span>
<span class="co">#&gt;  [50,]  4 0.88870 0.046620</span>
<span class="co">#&gt;  [51,]  4 0.89360 0.044500</span>
<span class="co">#&gt;  [52,]  4 0.89830 0.042480</span>
<span class="co">#&gt;  [53,]  4 0.90290 0.040550</span>
<span class="co">#&gt;  [54,]  4 0.90720 0.038710</span>
<span class="co">#&gt;  [55,]  4 0.91130 0.036950</span>
<span class="co">#&gt;  [56,]  4 0.91520 0.035270</span>
<span class="co">#&gt;  [57,]  4 0.91900 0.033670</span>
<span class="co">#&gt;  [58,]  4 0.92260 0.032140</span>
<span class="co">#&gt;  [59,]  4 0.92600 0.030670</span>
<span class="co">#&gt;  [60,]  4 0.92930 0.029280</span>
<span class="co">#&gt;  [61,]  4 0.93240 0.027950</span>
<span class="co">#&gt;  [62,]  4 0.93540 0.026680</span>
<span class="co">#&gt;  [63,]  4 0.93820 0.025470</span>
<span class="co">#&gt;  [64,]  4 0.94100 0.024310</span>
<span class="co">#&gt;  [65,]  4 0.94360 0.023200</span>
<span class="co">#&gt;  [66,]  4 0.94610 0.022150</span>
<span class="co">#&gt;  [67,]  4 0.94840 0.021140</span>
<span class="co">#&gt;  [68,]  4 0.95070 0.020180</span>
<span class="co">#&gt;  [69,]  4 0.95290 0.019260</span>
<span class="co">#&gt;  [70,]  4 0.95500 0.018390</span>
<span class="co">#&gt;  [71,]  4 0.95700 0.017550</span>
<span class="co">#&gt;  [72,]  5 0.95890 0.016760</span>
<span class="co">#&gt;  [73,]  5 0.96070 0.015990</span>
<span class="co">#&gt;  [74,]  5 0.96240 0.015270</span>
<span class="co">#&gt;  [75,]  5 0.96410 0.014570</span>
<span class="co">#&gt;  [76,]  5 0.96570 0.013910</span>
<span class="co">#&gt;  [77,]  5 0.96720 0.013280</span>
<span class="co">#&gt;  [78,]  5 0.96860 0.012670</span>
<span class="co">#&gt;  [79,]  5 0.97000 0.012100</span>
<span class="co">#&gt;  [80,]  5 0.97130 0.011550</span>
<span class="co">#&gt;  [81,]  5 0.97260 0.011020</span>
<span class="co">#&gt;  [82,]  6 0.97380 0.010520</span>
<span class="co">#&gt;  [83,]  6 0.97500 0.010040</span>
<span class="co">#&gt;  [84,]  6 0.97610 0.009588</span>
<span class="co">#&gt;  [85,]  6 0.97720 0.009152</span>
<span class="co">#&gt;  [86,]  6 0.97820 0.008736</span>
<span class="co">#&gt;  [87,]  6 0.97920 0.008339</span>
<span class="co">#&gt;  [88,]  6 0.98010 0.007960</span>
<span class="co">#&gt;  [89,]  6 0.98100 0.007598</span>
<span class="co">#&gt;  [90,]  6 0.98180 0.007253</span>
<span class="co">#&gt;  [91,]  6 0.98260 0.006923</span>
<span class="co">#&gt;  [92,]  6 0.98340 0.006609</span>
<span class="co">#&gt;  [93,]  6 0.98420 0.006308</span>
<span class="co">#&gt;  [94,]  6 0.98490 0.006022</span>
<span class="co">#&gt;  [95,]  6 0.98550 0.005748</span>
<span class="co">#&gt;  [96,]  7 0.98620 0.005487</span>
<span class="co">#&gt;  [97,]  7 0.98680 0.005237</span>
<span class="co">#&gt;  [98,]  7 0.98740 0.004999</span>
<span class="co">#&gt;  [99,]  7 0.98800 0.004772</span>
<span class="co">#&gt; [100,]  7 0.98850 0.004555</span>
<span class="co">#&gt; [1] "done bootstrap 1"</span>

<span class="co">#prepare table input for sankey plot </span>

reformat_LASSO &lt;-<span class="cf">function</span>(<span class="dt">c_selectID =</span> <span class="ot">NULL</span>, <span class="dt">s_selectID =</span> <span class="ot">NULL</span>, <span class="dt">LSOLDA_dat =</span> <span class="ot">NULL</span>, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#7570b3"</span>){
  LASSO_out &lt;-<span class="st"> </span><span class="kw"><a href="../reference/summary_prediction_lasso.html">summary_prediction_lasso</a></span>(<span class="dt">LSOLDA_dat=</span>LSOLDA_dat, <span class="dt">nPredSubpop =</span> nPredSubpop)
  LASSO_out &lt;-<span class="kw">as.data.frame</span>(LASSO_out)
  temp_name &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">"LASSO for subpop"</span>, <span class="st">"C"</span>, LASSO_out<span class="op">$</span>names)
  temp_name &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">" in target mixedpop"</span>, <span class="st">"S"</span>, temp_name)
  LASSO_out<span class="op">$</span>names &lt;-temp_name
  source &lt;-<span class="kw">rep</span>(<span class="kw">paste0</span>(<span class="st">"C"</span>,c_selectID,<span class="st">"S"</span>,s_selectID), <span class="kw">length</span>(temp_name))
  LASSO_out<span class="op">$</span>Source &lt;-<span class="st"> </span>source
  LASSO_out<span class="op">$</span>Node &lt;-<span class="st"> </span>source
  LASSO_out<span class="op">$</span>Nodes_group &lt;-<span class="st"> </span><span class="kw">rep</span>(Nodes_group, <span class="kw">length</span>(temp_name))
  <span class="kw">colnames</span>(LASSO_out) &lt;-<span class="kw">c</span>(<span class="st">"Value"</span>, <span class="st">"Target"</span>, <span class="st">"Source"</span>, <span class="st">"Node"</span>, <span class="st">"NodeGroup"</span>)
  LASSO_out<span class="op">$</span>Value &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.vector</span>(LASSO_out<span class="op">$</span>Value))
  <span class="kw">return</span>(LASSO_out)
}

LASSO_C1S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">1</span>, <span class="dt">s_selectID =</span><span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat1, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#7570b3"</span>)

LASSO_C2S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">2</span>, <span class="dt">s_selectID =</span><span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat2, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#1b9e77"</span>)

LASSO_C3S2  &lt;-<span class="st"> </span><span class="kw">reformat_LASSO</span>(<span class="dt">c_selectID=</span><span class="dv">3</span>, <span class="dt">s_selectID =</span><span class="dv">2</span>, <span class="dt">LSOLDA_dat=</span>LSOLDA_dat3, 
                          <span class="dt">nPredSubpop =</span> row_cluster, <span class="dt">Nodes_group =</span> <span class="st">"#e7298a"</span>)


combined &lt;-<span class="st"> </span><span class="kw">rbind</span>(LASSO_C1S2,LASSO_C2S2,LASSO_C3S2 )
combined &lt;-<span class="st"> </span>combined[<span class="kw">is.na</span>(combined<span class="op">$</span>Value) <span class="op">!=</span><span class="st"> </span><span class="ot">TRUE</span>,]
combined_D3obj &lt;-<span class="kw">list</span>(<span class="dt">Nodes=</span>combined[,<span class="dv">4</span><span class="op">:</span><span class="dv">5</span>], <span class="dt">Links=</span>combined[,<span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">1</span>)])

<span class="kw">library</span>(networkD3)

Node_source &lt;-<span class="st"> </span><span class="kw">as.vector</span>(<span class="kw">sort</span>(<span class="kw">unique</span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source)))
Node_target &lt;-<span class="st"> </span><span class="kw">as.vector</span>(<span class="kw">sort</span>(<span class="kw">unique</span>(combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target)))
Node_all &lt;-<span class="kw">unique</span>(<span class="kw">c</span>(Node_source, Node_target))

<span class="co">#assign IDs for Source (start from 0)</span>
Source &lt;-combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source
Target &lt;-<span class="st"> </span>combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target

<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(Node_all)){
  Source[Source<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
  Target[Target<span class="op">==</span>Node_all[i]] &lt;-i<span class="op">-</span><span class="dv">1</span>
}

combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Source &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(Source)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>Target &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(Target)
combined_D3obj<span class="op">$</span>Links<span class="op">$</span>LinkColor &lt;-<span class="st"> </span>combined<span class="op">$</span>NodeGroup

<span class="co">#prepare node info </span>
node_df &lt;-<span class="kw">data.frame</span>(<span class="dt">Node=</span>Node_all)
node_df<span class="op">$</span>id &lt;-<span class="kw">as.numeric</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(Node_all)<span class="op">-</span><span class="dv">1</span>)))

<span class="kw">suppressMessages</span>(<span class="kw">library</span>(dplyr))
Color &lt;-<span class="st"> </span>combined <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/tally.html">count</a></span>(Node, <span class="dt">color=</span>NodeGroup) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="dv">2</span>)
node_df<span class="op">$</span>color &lt;-<span class="st"> </span>Color<span class="op">$</span>color

<span class="kw">suppressMessages</span>(<span class="kw">library</span>(networkD3))
p1&lt;-<span class="kw"><a href="http://www.rdocumentation.org/packages/networkD3/topics/sankeyNetwork">sankeyNetwork</a></span>(<span class="dt">Links =</span>combined_D3obj<span class="op">$</span>Links, <span class="dt">Nodes =</span> node_df,  <span class="dt">Value =</span> <span class="st">"Value"</span>, <span class="dt">NodeGroup =</span><span class="st">"color"</span>, <span class="dt">LinkGroup =</span> <span class="st">"LinkColor"</span>, <span class="dt">NodeID=</span><span class="st">"Node"</span>, <span class="dt">Source=</span><span class="st">"Source"</span>, <span class="dt">Target=</span><span class="st">"Target"</span>, 
                  <span class="dt">fontSize =</span> <span class="dv">22</span> )
p1</code></pre></div>
<div id="htmlwidget-28662aba3f00bf408be2" style="width:700px;height:432.632880098888px;" class="sankeyNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-28662aba3f00bf408be2">{"x":{"links":{"source":[0,0,1,1,1,2,2],"target":[0,1,0,1,2,1,2],"value":[96.09375,3.25581395348837,14.453125,88.3720930232558,62.0689655172414,7.44186046511628,100],"group":["#7570b3","#7570b3","#1b9e77","#1b9e77","#1b9e77","#e7298a","#e7298a"]},"nodes":{"name":["C1S2","C2S2","C3S2"],"group":["#7570b3","#1b9e77","#e7298a"]},"options":{"NodeID":"Node","NodeGroup":"color","LinkGroup":"LinkColor","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":22,"fontFamily":null,"nodeWidth":15,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":true}},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#saveNetwork(p1, file = paste0(path,'Subpopulation_Net.html'))</span>
##R Setting Information
<span class="co">#sessionInfo()</span>
<span class="co">#rmarkdown::render("/Users/quan.nguyen/Documents/Powell_group_MacQuan/AllCodes/scGPS/vignettes/vignette.Rmd",html_document(toc = TRUE, toc_depth = 3))</span>
<span class="co">#rmarkdown::render("/Users/quan.nguyen/Documents/Powell_group_MacQuan/AllCodes/scGPS/vignettes/vignette.Rmd",pdf_document(toc = TRUE, toc_depth = 3))</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Install release version from CRAN</span>
<span class="co">#install.packages("pkgdown")</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation-instruction">1. Installation instruction</a></li>
      <li>
<a href="#a-simple-workflow-of-the-scgps">2. A simple workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#setup-scgps-objects">2.1 Setup scGPS objects</a></li>
      <li><a href="#run-predictions">2.2 Run predictions</a></li>
      <li><a href="#summarise-results">2.3 Summarise results</a></li>
      </ul>
</li>
      <li>
<a href="#a-complete-workflow-of-the-scgps">3. A complete workflow of the scGPS:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#identify-clusters-in-a-using-core">3.1 Identify clusters in a using CORE</a></li>
      <li><a href="#visualise-all-cluster-results-in-all-iterations">3.2 Visualise all cluster results in all iterations</a></li>
      <li><a href="#plot-the-optimal-clustering-result">3.3 Plot the optimal clustering result</a></li>
      <li><a href="#compare-clustering-results-with-other-dimensional-reduction-methods-e-g--cidr">3.4 Compare clustering results with other dimensional reduction methods (e.g., CIDR)</a></li>
      <li><a href="#find-gene-markers-and-annotate-clusters">3.5 Find gene markers and annotate clusters</a></li>
      </ul>
</li>
      <li>
<a href="#relationship-between-clusters-within-one-sample-or-between-two-samples">4. Relationship between clusters within one sample or between two samples</a><ul class="nav nav-pills nav-stacked">
<li><a href="#start-the-scgps-prediction-to-find-relationship-between-clusters">4.1 Start the scGPS prediction to find relationship between clusters</a></li>
      <li><a href="#display-summary-results-for-the-prediction">4.2 Display summary results for the prediction</a></li>
      <li><a href="#plot-the-relationship-between-clusters">4.3 Plot the relationship between clusters</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Quan Nguyen, Anne Asenbouth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
